---
title: "Auscad update 2022"
author: "Jen Li"
output: nil
---

# Set up for AUSCAD analysis
```{r, echo=FALSE, results = FALSE, warning = FALSE}
setwd("~/Documents/Project data - R files/auscad/")
  knitr::opts_chunk$set(echo = FALSE, cache = FALSE, message = FALSE, warning = FALSE)
  set.seed(42)

  source("data/setup.auscad.R")
    # # RUN THIS KNIT for clean.data.R if needed
      # source("data/clean.data.R")
  
  writeLines(capture.output(sessionInfo()),"data/current.packages.txt")
  write.table(pkg.versions, "data/package.versions.txt", append = FALSE,   
              sep = " ", dec = ".",row.names = TRUE, col.names = TRUE)

# My data organised in the following way: metadata = clinical metadata, samples = sequencing metadata,  counts = raw counts matrix,  conversions = ensembl, entrez, symbols. Keep original so not overwritten
  auscad.sorted = new("auscad.sorted",
                      metadata.all = readRDS("data/annotatedDB.RDS"),
                      samples.all = readRDS("data/samples.all.RDS"),
                      counts.all = readRDS("data/counts.all.RDS"),
                      conversions = readRDS("data/conversion.RDS"),
                      samples.biopsy.0m = readRDS("data/biopsy0m.RDS"),
                      counts.biopsy.0m = readRDS("data/counts0m.RDS"),
                      samples.biopsy.1m = readRDS("data/biopsy1m.RDS"),
                      counts.biopsy.1m = readRDS("data/counts1m.RDS"),
                      samples.biopsy.3m = readRDS("data/biopsy3m.RDS"),
                      counts.biopsy.3m = readRDS("data/counts3m.RDS"),
                      samples.blood.0m = readRDS("data/blood0m.RDS"),
                      counts.blood.0m = readRDS("data/bloodcounts0m.RDS"),
                      samples.blood.3m = readRDS("data/blood3m.RDS"),
                      counts.blood.3m = readRDS("data/bloodcounts3m.RDS"))

# Variables/change levels etc of clin data
dgf.filter = auscad.sorted@metadata.all
dgf.filter$graft_number = as.factor(dgf.filter$graft_number)
dgf.filter$dgf_hd_days = as.numeric(ifelse(is.na(dgf.filter$dgf_hd_days),
                                  0,  dgf.filter$dgf_hd_days))

# Make matrix with only deceased donor KTx
dktx.only = dgf.filter[dgf.filter$organ_type == "Kidney Transplant",]

# Number of biopsy RNAseq available at 0m
gene.check = auscad.sorted@samples.biopsy.0m
gene.check$dgf_hd_criteria = factor(gene.check$dgf_hd_criteria, 
                                    levels = c(0,1), 
                                    labels = c("Control", "DGF.HD"))
gene.check = gene.check %>% 
  mutate(dgf_diff_criteria = as.factor(case_when(
    dgf_hd_criteria == "DGF.HD" ~ "DGF.HD",
    dgf_cr_criteria1 == "DGF.Cr" ~ "DGF.Cr",
    dgf_hd_criteria != "DGF.HD" & dgf_cr_criteria1 != "DGF.Cr" ~ "control")))
```

### Cibersort: cibersort.stanford.edu.au, login jennifer.li1, pw Sortcells2022

# Set up the groups and conditions for DGE comparison
```{r, echo=FALSE, results = FALSE, warning = FALSE}

# 0m, DGF combined vs nil
criteria = "dgf_combined1"
samples = auscad.sorted@samples.biopsy.0m
counts = auscad.sorted@counts.biopsy.0m

# 1m, DGF combined vs nil
criteria = "dgf_combined1"
samples = auscad.sorted@samples.biopsy.1m
samples = samples %>% filter(bpar_0_1m == "No BPAR")
samples = samples %>% filter(subclin_1m == "No Subclin")

counts = auscad.sorted@counts.biopsy.1m
counts = counts[,colnames(counts) %in% rownames(samples)]

# 0m, DGF HD only
criteria = "dgf_hd_criteria1"
counts = auscad.sorted@counts.biopsy.0m
samples = auscad.sorted@samples.biopsy.0m

# 0m, DGF HD only but split days +/- 7
criteria = "dgf_hd_days1"
samples = auscad.sorted@samples.biopsy.0m
samples = samples %>% filter(dgf_hd_days>0)
sum((samples$dgf_hd_days>=7) == TRUE)
sum((samples$dgf_hd_days<7) == TRUE)
samples$dgf_hd_days1 = ifelse(samples$dgf_hd_days>=7, "longHD", "shortHD")
samples$dgf_hd_days1 = as.factor(samples$dgf_hd_days1)
levels(samples$dgf_hd_days1 )

counts = auscad.sorted@counts.biopsy.0m
counts = counts[,colnames(counts) %in% rownames(samples)]
```


# Normalise data, check and remove batch effects
```{r, echo=FALSE, results = FALSE, warning = FALSE}

check = sort.normalise(samples, 
                       counts , 
                       criteria)

summary(check$samples$group)

dge.lcpm = edgeR::cpm(check, log=TRUE)
  
  boxplot(dge.lcpm, las=2, col=brewer.pal(ncol(dge.lcpm), "Paired"), 
          ylab = "Expression in log-cpm",
          main="log-CPM After Filtering & TMM")
  
  pheatmap(cor(getCounts(check)),
         annotation = dplyr::select(check$samples, group),
         show_rownames = FALSE)

check.table = as.data.frame(t(dge.lcpm))
check.table$Condition = check$samples$group
check.table$Batch = check$samples$batch
  pca = prcomp(as.data.frame(t(dge.lcpm)), scale = TRUE)
  p1 = autoplot(pca, data=check.table, colour = "Condition", size = 5,
               frame = TRUE, frame.type = 'norm') + theme_classic()
  p2 = autoplot(pca, data=check.table, colour = "Batch", size = 5,
               frame = TRUE, frame.type = 'norm') + theme_classic()
  
check.table1 = as.data.frame(t(removeBatchEffect(dge.lcpm, check$samples$batch)))
check.table1$Condition = check$samples$group
check.table1$Batch = check$samples$batch
  p3 = autoplot(prcomp(check.table1[,-((ncol(check.table1)-1):ncol(check.table1))],
                       scale = TRUE, center = TRUE),
         data=check.table1, colour = "Batch", size = 5,
         frame = TRUE, frame.type = 'norm') + theme_classic()
  ggarrange(p2,p3,ncol = 2, nrow = 1)

d.pca = data.frame(PC_1 = pca$x[,1],PC_2 = pca$x[,2])
d.pca = cbind(d.pca, check$samples)
fit1 = lm(PC_1 ~ group + batch + lib.size + donor_age + donor_gender + gender +  cit_min, data = d.pca)
fit2 = glm(group ~ batch + lib.size +donor_age + donor_gender +  gender  +  cit_min, data = d.pca, family = "binomial")
summary(fit1)
summary(fit2)

  ggplot(d.pca, aes(x = PC_1, y = PC_2, color = dgf_combined, text = paste(samplename,batch, sep = "__"))) +
    geom_point(size = 4) + theme_bw()
```

# DEGs - for dgf_combined1
```{r, echo=FALSE, results = FALSE, warning = FALSE}
design = model.matrix(~0 + group + batch, data = check$samples)
  colnames(design) = gsub("batch", "", colnames(design))
  colnames(design) =  gsub("group", "", colnames(design))
  colnames(design) =  gsub(" ", "", colnames(design))
auscad.0m = check[,rownames(design)]
design = design[,which(colSums(design) > 2)]

auscad.0m = estimateDisp(auscad.0m, design)
glmfit.0m =  glmQLFit(auscad.0m, design)
con.mat = makeContrasts(DGF = DGF - Control, levels = design)

# EdgeR
fit.0m = glmLRT(glmfit.0m, contrast = con.mat[,"DGF"])
top.tags.0m = topTags(fit.0m,  n = Inf)
summary(decideTests(fit.0m, adjust.method = "fdr", p.value = 0.05))
top.tags.0m = as.data.frame(top.tags.0m) %>% mutate(Symbol = rownames(top.tags.0m))
top.tags.0m = top.tags.0m %>% mutate(Significant = FDR < 0.05)

  plotBCV(auscad.0m)
  plotMD(fit.0m)
  
  ggplot(top.tags.0m %>% arrange(Significant),
         aes(x = logFC, y = -log(PValue), color = Significant,
             text = paste("Name:", top.tags.0m$Symbol))) + geom_point() + theme_bw()

# Limma/ebayes  
dds = voom(check, design, plot=TRUE)
fit = contrasts.fit(lmFit(dds, design), con.mat)
efit = eBayes(fit, robust = TRUE)
  plotSA(efit, main="Final model: Mean-variance trend")
results = decideTests(efit, adjust.method = "BH", p.value = 0.05)
top.tags.limma = topTable(efit, adjust.method="fdr", sort.by = "B", n = Inf)
top.tags.limma = top.tags.limma %>% filter(adj.P.Val < 0.05)
```

# DEGs - for dgf_hd_criteria
```{r, echo=FALSE, results = FALSE, warning = FALSE}
design = model.matrix(~0 + group + batch, data = check$samples)
  colnames(design) = gsub("batch", "", colnames(design))
  colnames(design) =  gsub("group", "", colnames(design))
  colnames(design) =  gsub(" ", "", colnames(design))
  colnames(design) =  gsub("DGF:HDcriteria", "DGF", colnames(design))
auscad.0m = check[,rownames(design)]
design = design[,which(colSums(design) > 2)]

auscad.0m = estimateDisp(auscad.0m, design)
glmfit.0m =  glmQLFit(auscad.0m, design)
con.mat = makeContrasts(DGF = DGF - Control, levels = design)

# EdgeR
fit.0m = glmLRT(glmfit.0m, contrast = con.mat[,"DGF"])
top.tags.0m = topTags(fit.0m,  n = Inf)
summary(decideTests(fit.0m, adjust.method = "fdr", p.value = 0.05))
top.tags.0m = as.data.frame(top.tags.0m) %>% mutate(Symbol = rownames(top.tags.0m))
top.tags.0m = top.tags.0m %>% mutate(Significant = FDR < 0.05)

  plotBCV(auscad.0m)
  plotMD(fit.0m)
  
  ggplot(top.tags.0m %>% arrange(Significant),
         aes(x = logFC, y = -log(PValue), color = Significant,
             text = paste("Name:", top.tags.0m$Symbol))) + geom_point() + theme_bw()

# Limma/ebayes  
dds = voom(check, design, plot=TRUE)
fit = contrasts.fit(lmFit(dds, design), con.mat)
efit = eBayes(fit, robust = TRUE)
  plotSA(efit, main="Final model: Mean-variance trend")
results = decideTests(efit, adjust.method = "BH", p.value = 0.05)
top.tags.limma = topTable(efit, adjust.method="fdr", sort.by = "B", n = Inf)
top.tags.limma = top.tags.limma %>% filter(adj.P.Val < 0.05)
```

# DEGs - for dgf_hd__criteria_days >=7
```{r, echo=FALSE, results = FALSE, warning = FALSE}
design = model.matrix(~0 + group + batch, data = check$samples)
  colnames(design) = gsub("batch", "", colnames(design))
  colnames(design) =  gsub("group", "", colnames(design))
  colnames(design) =  gsub(" ", "", colnames(design))
auscad.0m = check[,rownames(design)]
design = design[,which(colSums(design) > 2)]

auscad.0m = estimateDisp(auscad.0m, design)
glmfit.0m =  glmQLFit(auscad.0m, design)
con.mat = makeContrasts(DGF.Severity = longHD - shortHD, levels = design)

# EdgeR
fit.0m = glmLRT(glmfit.0m, contrast = con.mat[,"DGF.Severity"])
top.tags.0m = topTags(fit.0m,  n = Inf)
summary(decideTests(fit.0m, adjust.method = "fdr", p.value = 0.05))
top.tags.0m = as.data.frame(top.tags.0m) %>% mutate(Symbol = rownames(top.tags.0m))
top.tags.0m = top.tags.0m %>% mutate(Significant = FDR < 0.05)

  plotBCV(auscad.0m)
  plotMD(fit.0m)
  
  ggplot(top.tags.0m %>% arrange(Significant),
         aes(x = logFC, y = -log(PValue), color = Significant,
             text = paste("Name:", top.tags.0m$Symbol))) + geom_point() + theme_bw()

# Limma/ebayes  
dds = voom(check, design, plot=TRUE)
fit = contrasts.fit(lmFit(dds, design), con.mat)
efit = eBayes(fit, robust = TRUE)
  plotSA(efit, main="Final model: Mean-variance trend")
results = decideTests(efit, adjust.method = "BH", p.value = 0.05)
top.tags.limma = topTable(efit, adjust.method="fdr", sort.by = "B", n = Inf)
top.tags.limma = top.tags.limma %>% filter(adj.P.Val < 0.05)
```

# GSEA with GO ontology
```{r, fig.height=8, fig.width=16}
organism = org.Hs.eg.db

  Symbol = mapIds(org.Hs.eg.db, keys = rownames(top.tags.0m), keytype = "ENSEMBL",
                   column = "SYMBOL", multiVals = "first")
  keep1 = which(!is.na(Symbol) & !duplicated(Symbol))
  Symbol.keep <- Symbol[keep1]
  top.tags.0m <- top.tags.0m[keep1,]
  top.tags.0m$symbol = Symbol.keep
  
# FC 1.5
Res = top.tags.0m[abs(top.tags.0m$logFC) > log2(1.5),]$logFC
names(Res) = top.tags.0m[abs(top.tags.0m$logFC) > log2(1.5),]$symbol
Res = sort(Res, decreasing = TRUE)

# FC 1
Res = top.tags.0m[abs(top.tags.0m$logFC) > log2(1),]$logFC
names(Res) = top.tags.0m[abs(top.tags.0m$logFC) > log2(1),]$symbol
Res = sort(Res, decreasing = TRUE)

gse <- gseGO(geneList=Res, 
             ont ="ALL", 
             keyType = "SYMBOL",
             minGSSize = 3, 
             maxGSSize = 8000,
             eps = 0,
             pvalueCutoff = 0.01, 
             verbose = TRUE,
             seed = 23,
             OrgDb = organism)


  ggo.bp = filter(gse, p.adjust < 0.05, qvalues < 0.05)
  x2.bp = pairwise_termsim(ggo.bp)
  
  ggo.bp1 = as.data.frame(ggo.bp)
  ggo.bp1 = ggo.bp1 %>% dplyr::mutate(log.p.adj = -log(p.adjust)) %>% arrange(ggo.bp1, NES)
  ggo.bp1$Pathway.Direction = cut(-1*ggo.bp1$NES, c(-Inf,0,Inf), c("Upregulated", "Suppressed"))
  ggo.bp1$log.p.adj = ifelse(ggo.bp1$Pathway.Direction == "Suppressed", 
                             -1*ggo.bp1$log.p.adj, ggo.bp1$log.p.adj)

  colnames(ggo.bp1)[4] = "Count"
  ggo.bp1 = removecrap(ggo.bp1)
  
  ggo.bp1 = pickrelevant(ggo.bp1)
  ggo.bp1 = trimmer(ggo.bp1)
  
  ggo.bp2 = trimmer(ggo.bp1)
  
  pg1.bp = dotplot(ggo.bp1, showCategory = 20, title = "GSEA-GO", split = ".sign") + facet_grid(.~.sign)
  pg2.bp = gseaplot(ggo.bp1, by = "all", geneSetID = 1)
  pg3.bp = ridgeplot(ggo.bp1) + labs(x = "enrichment distribution")
  pg4.bp = emapplot(x2.bp)
  pg5.bp = ggplot(ggo.bp1, aes(x = log.p.adj, y = Description)) + 
              geom_point(aes(color= p.adjust, size = Count)) + theme_bw() + 
              scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) + 
              theme(axis.text.y = element_text(size = 10)) +xlab("-log10 of P-value") +
              ylab("Description") + 
              ggtitle("GSEA-GO:BP") + 
              facet_grid(~Pathway.Direction)
  pg6.bp = cnetplot(x2.bp, categorySize = "adj.P.Val", 
                    colorEdge = TRUE, foldChange = top.tags.limma$logFC, 
                    node_label = "all", cex_label_category = 1.3)
  
ggplot(ggo.bp1, aes(x = log.p.adj, y = Description)) + 
              geom_point(aes(color= p.adjust, size = Count)) + theme_bw() + 
              scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) + 
              theme(axis.text.y = element_text(size = 10)) +xlab("-log10 of P-value") +
              ylab("Description") + 
              ggtitle("GSEA-GO:BP") + 
              facet_wrap(group~Pathway.Direction)

ggplot(filter(ggo.bp1 ,group == "innate"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(0,50)+
              geom_vline(xintercept=0) + facet_wrap(Pathway.Direction~.)
 
g1 = ggplot(filter(ggo.bp1 ,group == "innate"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+xlim(-20,20)+
              geom_vline(xintercept=0) + facet_grid(group~.)
  
g2 = ggplot(filter(ggo.bp1 ,group == "adaptive"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+xlim(-20,20)+
              geom_vline(xintercept=0) + facet_grid(group~.)

g3 = ggplot(filter(ggo.bp1 ,group == "death"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(-20,20)+
              geom_vline(xintercept=0) + facet_grid(group~.)

g4 = ggplot(filter(ggo.bp1 ,group == "metab"),
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(-20,20)+
              geom_vline(xintercept=0) + facet_grid(group~.)

g5 = ggplot(filter(ggo.bp1 ,group == "cell"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+xlim(-20,20)+
              geom_vline(xintercept=0) + facet_grid(group~.)

g6 = ggplot(filter(ggo.bp1 ,group == "death"| group == "innate"),  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(0,50)+
              geom_vline(xintercept=0)

tiff("figures/dgf_combined_0m.fc1.5.tiff", units = "cm", res = 300, height = 28, width = 32)
ggplot(ggo.bp1,  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(-1,20)+
              geom_vline(xintercept=0)
dev.off()

write.xlsx(ggo.bp1, "csv/dgf_combined_0m_fc1.5.xlsx")

tiff("figures/dgf_combined_1m.fc1.5.tiff", units = "cm", res = 300, height = 20, width = 35)
ggplot(ggo.bp1,  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(-20,20)+
              geom_vline(xintercept=0)
dev.off()
write.xlsx(ggo.bp1, "csv/dgf_combined_1m_fc1.5.xlsx")

tiff("figures/dgf_HD_0m.fc1.5.tiff", units = "cm", res = 300, height = 28, width = 32)
ggplot(ggo.bp1,  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+ xlim(-1,15)+
              geom_vline(xintercept=0)
dev.off()
write.xlsx(ggo.bp1, "csv/dgf_HD_0m_fc1.5.xlsx")

tiff("figures/dgf_HD_7days.fc1.tiff", units = "cm", res = 300, height = 28, width = 32)
ggplot(ggo.bp2,  
              aes(y= Description, x = log.p.adj, fill = Pathway.Direction))+
              geom_bar(stat = "identity", colour = "black", size = 0.2) +
              theme(axis.text.y = element_text(size = 12))+
              geom_vline(xintercept=0)
dev.off()
write.xlsx(ggo.bp2, "csv/dgf_HD_7days.xlsx")

```



# This section - move to genomics side
### [PICK ONE] Set up the groups and conditions for DGE comparison
### Bulk deconvolution
Can use CibersortX or Music, similar. If unknown cells (like tumour) - Edge is likely to work better
```{r, echo=TRUE, results = FALSE, warning = FALSE, eval = FALSE}
# 0m, DGF combined vs nil
criteria = "dgf_combined1"
samples = auscad.sorted@samples.biopsy.0m
counts = auscad.sorted@counts.biopsy.0m

# 1m, DGF combined vs nil
criteria = "dgf_combined1"
samples = auscad.sorted@samples.biopsy.1m
samples = samples %>% filter(bpar_0_1m == "No BPAR")
samples = samples %>% filter(subclin_1m == "No Subclin")

counts = auscad.sorted@counts.biopsy.1m
counts = counts[,colnames(counts) %in% rownames(samples)]

# 0m, DGF HD only
criteria = "dgf_hd_criteria1"
counts = auscad.sorted@counts.biopsy.0m
samples = auscad.sorted@samples.biopsy.0m

# 0m, DGF HD only but split days +/- 7
criteria = "dgf_hd_days1"
samples = auscad.sorted@samples.biopsy.0m
samples = samples %>% filter(dgf_hd_days>0)
sum((samples$dgf_hd_days>=7) == TRUE)
sum((samples$dgf_hd_days<7) == TRUE)
samples$dgf_hd_days1 = ifelse(samples$dgf_hd_days>=7, "longHD", "shortHD")
samples$dgf_hd_days1 = as.factor(samples$dgf_hd_days1)
levels(samples$dgf_hd_days1 )

counts = auscad.sorted@counts.biopsy.0m
counts = counts[,colnames(counts) %in% rownames(samples)]
```







# DESeq2
```{r}
dds = DESeqDataSetFromMatrix(countData = k.0ma, 
                             colData = metadata, 
                             design =~ batch + gender1 + dgf_combined1)

dds = DESeq(dds)
res = results(dds, alpha = 0.05, lfcThreshold = log(1.5))
rld = vst(dds) # use rlog when not such big set

head(res)
resultsNames(dds)
levels(dds$dgf_combined1)
# plot.disp = plotDispEsts(dds, main="dispersion plot")
# topgenes=head(order(-rowVars(assay(rld))),100)

DESeq2::plotPCA(rld, intgroup = "dgf_combined1") + theme_bw() + geom_point(size = 2)
DESeq2::plotMA(dds)

select = order(rowMeans(counts(dds,normalized=TRUE)), decreasing=TRUE)
ntd = normTransform(dds)
df = as.data.frame(colData(dds)[,c("dgf_combined1")])
rownames(df) = colnames(assay(ntd))
colnames(df) = "DGF status"

# select the 'contrast' you want
annotation_data <- as.data.frame(colData(rld)[c("ConditionA","ConditionB")])
pheatmap(matrix, annotation_col=annotation_data)

cal_z_score = function(x){(x - mean(x)) / sd(x)}
ntd_z = data.frame(t(apply(assay(ntd), 1, cal_z_score)))

# Set up for pheatmap
ntd_z.symbol = ntd_z
ensemblid = rownames(ntd_z.symbol)
samples.ntd = colnames(ntd_z.symbol)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
ntd_z.symbol = cbind(ntd_z.symbol, symbol=eID)
ntd_z.symbol <- ntd_z.symbol[!duplicated(ntd_z.symbol$symbol),]
ntd_z.symbol <- na.omit(ntd_z.symbol)
rownames(ntd_z.symbol) = ntd_z.symbol$symbol
ntd_z.symbol = ntd_z.symbol[,1:(ncol(ntd_z.symbol) - 1)]
ntd_z.symbol = data.frame(t(ntd_z.symbol))

pheatmap::pheatmap(ntd_z.symbol[,1:20], annotation_row = df, 
                   color = colorRampPalette(c("blue", "navy", "white", "tan2", "orange"))(10))

jpeg("figures/pheatmap1.jpeg", width = 1200, height = 1000)
pheatmap::pheatmap(ntd_z.symbol[,1:100], annotation_row = df, 
                   color = colorRampPalette(c("blue", "navy", "white", "tan2", "orange"))(10))
dev.off()

res = results(dds, alpha = 0.05, name = "dgf_combined1_DGF_vs_Control")
ensemblid = rownames(res)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res = cbind(res, symbol=eID)
res = res[!duplicated(res$symbol),]
res = na.omit(res)
res = data.frame(res)
res = res[abs(res$log2FoldChange) >= 1.5,]
res = res[order(res$padj),]
  
write.xlsx(res, "deseq2.dgf.xlsx", sheetName = "DGF.vs.control", col.names = TRUE, row.names = TRUE)

res


match1 = assay(rld)[rownames(res),]
match1 = data.frame(match1)
ensemblid = rownames(match1)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
match1 = cbind(match1, symbol=eID)
match1 <- match1[!duplicated(match1$symbol),]
match1 <- na.omit(match1)
rownames(match1) = match1$symbol
match1 = match1[,1:(ncol(match1) - 1)]
match1 = data.frame(t(match1))
pheatmap::pheatmap(match1, 
                   color = colorRampPalette(c("blue", "navy", "white", "tan2", "orange"))(10))
```

# Re-run deseq to change the design +/- contrast
```{r}
dds1 = DESeqDataSetFromMatrix(countData = k.0ma, 
                             colData = metadata, 
                             design =~ batch + gender1 + dgf_severity)

dds1 = DESeq(dds1)
res1 = results(dds1, alpha = 0.05, lfcThreshold = 1.5)
rld1 = vst(dds1) # use rlog when not such big set

select1 = order(rowMeans(counts(dds1,normalized=TRUE)), decreasing=TRUE)
ntd1 = normTransform(dds1)
df1 = as.data.frame(colData(dds1)[,c("dgf_severity")])
rownames(df1) = colnames(assay(ntd1))
colnames(df1) = "DGF status"

cal_z_score = function(x){(x - mean(x)) / sd(x)}
ntd_z1 = data.frame(t(apply(assay(ntd1), 1, cal_z_score)))

# Set up for pheatmap
ntd_z.symbol1 = ntd_z1
ensemblid = rownames(ntd_z.symbol1)
samples.ntd = colnames(ntd_z.symbol1)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
ntd_z.symbol1 = cbind(ntd_z.symbol1, symbol=eID)
ntd_z.symbol1 <- ntd_z.symbol1[!duplicated(ntd_z.symbol1$symbol),]
ntd_z.symbol1 <- na.omit(ntd_z.symbol1)
rownames(ntd_z.symbol1) = ntd_z.symbol1$symbol
ntd_z.symbol1 = ntd_z.symbol1[,1:(ncol(ntd_z.symbol1) - 1)]
ntd_z.symbol1 = data.frame(t(ntd_z.symbol1))

jpeg("figures/pheatmap2.jpeg", width = 1200, height = 1000)
pheatmap::pheatmap(ntd_z.symbol1[,1:100], annotation_row = df1, 
                   color = colorRampPalette(c("blue", "navy", "white", "tan2", "orange"))(10))
dev.off()


res1a = results(dds1, alpha = 0.05, name = resultsNames(dds1)[7])
ensemblid = rownames(res1a)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res1a = cbind(res1a, symbol=eID)
res1a = res1a[!duplicated(res1a$symbol),]
res1a = na.omit(res1a)
res1a = data.frame(res1a)
res1a = res1a[abs(res1a$log2FoldChange) >= 1.5,]
res1a = res1a[order(res1a$padj),]

res1b = results(dds1, alpha = 0.05, name = resultsNames(dds1)[8])
ensemblid = rownames(res1b)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res1b = cbind(res1b, symbol=eID)
res1b = res1b[!duplicated(res1b$symbol),]
res1b = na.omit(res1b)
res1b = data.frame(res1b)
res1b = res1b[abs(res1b$log2FoldChange) >= 1.5,]
res1b = res1b[order(res1b$padj),]

res1c = results(dds1, alpha = 0.05, name = resultsNames(dds1)[9])
ensemblid = rownames(res1c)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res1c = cbind(res1c, symbol=eID)
res1c = res1c[!duplicated(res1c$symbol),]
res1c = na.omit(res1c)
res1c = data.frame(res1c)
res1c = res1c[abs(res1c$log2FoldChange) >= 1.5,]
res1c = res1c[order(res1c$padj),]



match1 = assay(rld1)[rownames(res1a),]
match1 = data.frame(match1)
ensemblid = rownames(match1)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
match1 = cbind(match1, symbol=eID)
match1 <- match1[!duplicated(match1$symbol),]
match1 <- na.omit(match1)
rownames(match1) = match1$symbol
match1 = match1[,1:(ncol(match1) - 1)]
match1 = data.frame(t(match1))


pheatmap::pheatmap(match1, fontsize_row = 6, fontsize_col = 10,
                   color = colorRampPalette(c("blue", "navy", "white", "tan2", "orange"))(10))


write.xlsx(res1a, "deseq2.dgf.xlsx", sheetName = "DGF.Cr.vs.control", 
           col.names = TRUE, row.names = TRUE, append = TRUE)
write.xlsx(res1b, "deseq2.dgf.xlsx", sheetName = "DGF.HDmin.vs.control", 
           col.names = TRUE, row.names = TRUE, append = TRUE)
write.xlsx(res1c, "deseq2.dgf.xlsx", sheetName = "DGF.HDmax.vs.control", 
           col.names = TRUE, row.names = TRUE, append = TRUE)
```

```{r}
pcaData1 <- plotPCA(rld, intgroup=c("dgf_hd_criteria1", "dgf_combined1", "dgf_cr_criteria",
                                     "donor_criteria",  "bpar_any", "organ_type1"), returnData=TRUE)

pcaData2 <- plotPCA(rld1, intgroup=c("dgf_severity", "dgf_combined1", "dgf_cr_criteria",
                                     "donor_criteria",  "bpar_any", "organ_type1"), returnData=TRUE)


jpeg(filename = "figures/pca.facet1.jpeg", width = 600, height = 400)
ggplot(pcaData1, aes(PC1, PC2, color = metadata$donor_criteria)) + theme_bw() + geom_point(size=1.5) + 
         facet_grid(metadata$dgf_combined1 + metadata$dgf_hd_criteria1 ~ metadata$bpar_any1)
dev.off()

jpeg(filename = "figures/pca.facet2.jpeg", width = 500, height = 400)
ggplot(pcaData2, aes(PC1, PC2, colour = metadata$donor_criteria)) + theme_bw() + geom_point(size=1.5) + 
         facet_grid(metadata$dgf_severity ~ metadata$bpar_any1)
dev.off()

plotMDS(dds, col = factor(metadata$dgf_combined1))


```

```{r}
# Volcano plot version 1, default cutoffs for log2FC is 2, p=10e-6
# EnhancedVolcano(res,lab=rownames(res),  x='log2FoldChange', y='pvalue')

resultsNames(dds)
res.a = data.frame(results(dds, alpha = 0.05, name = "dgf_combined1_DGF_vs_Control"))
ensemblid = rownames(res.a)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res.a = cbind(res.a, symbol=eID)
res.a <- res.a[!duplicated(res.a$symbol),]
res.a <- na.omit(res.a)
rownames(res.a) = res.a$symbol

jpeg("figures/volc1.jpg", width = 600, height = 600)
EnhancedVolcano(res.a,
    lab = res.a$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'DGF: all criteria',
    pCutoff = 0.05,
    FCcutoff = log(1.5),
    pointSize = 3.0,
    labSize = 3.0,
    xlim= c(min(res.a$log2FoldChange)-1, max(res.a$log2FoldChange)+1),
    ylim= c(min(-log10(res.a$pvalue))-0.5, max(-log10(res.a$log2FoldChange))+1.5),
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'bottom',
    legendLabSize = 10,
    legendIconSize = 3.0)
dev.off()

  
  

```

```{r}
# Volcano plot version 1, default cutoffs for log2FC is 2, p=10e-6
# EnhancedVolcano(res,lab=rownames(res),  x='log2FoldChange', y='pvalue')

res.b = data.frame(results(dds1, alpha = 0.05, name = resultsNames(dds1)[9]))
ensemblid = rownames(res.b)
eID = mapIds(org.Hs.eg.db, keys = as.character(ensemblid), 
             column = c("SYMBOL"), keytype = "ENSEMBL", multiVals = "first")
res.b = cbind(res.b, symbol=eID)
res.b <- res.b[!duplicated(res.b$symbol),]
res.b <- na.omit(res.b)
rownames(res.b) = res.b$symbol

jpeg("figures/volc2 Cr vs Cont.jpg", width = 600, height = 600)
EnhancedVolcano(res.b,
    lab = res.b$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'DGF: Cr criteria',
    pCutoff = 0.05,
    FCcutoff = 1.5,
    pointSize = 3.0,
    labSize = 3.0,
    xlim= c(min(res.b$log2FoldChange)-1, max(res.b$log2FoldChange)+1),
    ylim= c(min(-log10(res.b$pvalue))-0.5, max(-log10(res.b$log2FoldChange))+1.5),
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'bottom',
    legendLabSize = 10,
    legendIconSize = 3.0)
dev.off()

jpeg("figures/volc3 HDmin vs Cont.jpg", width = 600, height = 600)
EnhancedVolcano(res.b,
    lab = res.b$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'DGF: HD (<7d) vs control',
    pCutoff = 0.05,
    FCcutoff = 1.5,
    pointSize = 3.0,
    labSize = 3.0,
    xlim= c(min(res.b$log2FoldChange)-1, max(res.b$log2FoldChange)+1),
    ylim= c(min(-log10(res.b$pvalue))-0.5, max(-log10(res.b$log2FoldChange))+1.5),
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'bottom',
    legendLabSize = 10,
    legendIconSize = 3.0)
dev.off()
  
jpeg("figures/volc3 HDmax vs Cont.jpg", width = 600, height = 600)
EnhancedVolcano(res.b,
    lab = res.b$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'DGF: HD (>7d) vs control',
    pCutoff = 0.05,
    FCcutoff = 1.5,
    pointSize = 3.0,
    labSize = 3.0,
    xlim= c(min(res.b$log2FoldChange)-1, max(res.b$log2FoldChange)+1),
    ylim= c(min(-log10(res.b$pvalue))-0.5, max(-log10(res.b$log2FoldChange))+1.5),
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'bottom',
    legendLabSize = 10,
    legendIconSize = 3.0)
dev.off()
```

```{r}
#plotcounts ######### ADD specific more ##########
d2<-plotCounts(dds2, gene="ENSG00000269981", intgroup=c("dgf_criteria"), returnData=TRUE)
plotCounts(dds2, gene="ENSG00000269981", intgroup="dgf_criteria")

ggplot(d2, aes(x = dgf_criteria, y = count, color = dgf_criteria)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  theme_bw() +
  ggtitle("ENSG00000269981") +
  theme(plot.title = element_text(hjust = 0.5))

## Examine plot of p-values
hist(resdgf$pvalue, breaks=50, col="grey")
attr(resdgf, "filterThreshold")

```

```{r}
res.path = res.a$log2FoldChange
names(res.path) = res.a$symbol

# Feed into GSEA:GO:BP
gse.temp = c()
gse.temp = gseGO(geneList=sort(res.path, decreasing = TRUE), 
             ont ="BP", 
             keyType = "SYMBOL",
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE,
             seed = 1234,
             OrgDb = org.Hs.eg.db)

gse = as.data.frame(gse.temp)
gse = gse %>% dplyr::mutate(log.p.adj = -log(p.adjust)) %>% arrange(gse, NES)
gse$Pathway.Direction = cut(-1*gse$NES, c(-Inf,0,Inf), c("Upregulated", "Suppressed"))
colnames(gse)[4] = "Count"
gse = filter(gse, gse$Description %in% remove == FALSE)
gse = filter(gse, gse$Description %in% immune == TRUE)
immune = c(
"activation of innate immune response"  ,"innate immune response activating cell surface receptor signaling pathway" , "neutrophil mediated immunity"   , "lymphocyte mediated immunity" , "humoral immune response mediated by circulating immunoglobulin"    ,"antigen processing and presentation of peptide antigen","acute inflammatory response", "innate immune response-activating signal transduction"   , "oxidative phosphorylation",    "mitochondrial electron transport, NADH to ubiquinone" , "kidney epithelium development" ,  "nephron tubule development"  , "regulation of NAD metabolic process"    )
ggplot(gse, aes(x = log.p.adj, y = Description)) + 
    geom_point(aes(color= p.adjust, size = Count)) + theme_bw() +
    scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) + 
  theme(axis.text.y = element_text(size = 10)) +xlab("-log10 of P-value") +
    ylab("Description") + ggtitle("GSEA-GO:BP") + facet_grid(~Pathway.Direction)



```

```{r}
remove = c("skeletal system development", "blood vessel endothelial cell proliferation involved in sprouting angiogenesis", "growth plate cartilage development", "defense response, response to external stimulus", "tissue development",  "response to organic substance" , "response to lithium ion" , "response to progesterone","Fc-gamma receptor signaling pathway","wound healing"   , "response to progesterone"  , "wound healing"   ,"response to chemical"    ,"interspecies interaction between organisms"  ,"negative regulation of myoblast differentiation" , "secretion" , "animal organ development" , "regulation of response to stimulus" , "positive regulation of response to stimulus" , "regulation of body fluid levels"  ,"regulation of secretion" , "positive regulation of secretion", "cartilage development"  , "spindle assembly" , "regulation of multicellular organismal process"  ,"positive regulation of multicellular organismal process", "spindle midzone assembly","mitotic spindle midzone assembly" , "negative regulation of cytoskeleton organization"  , "response to other organism" ,"sulfation" , "regulation of chromosome segregation" ,"endochondral bone morphogenesis", "cartilage development involved in endochondral bone morphogenesis"  ,  "connective tissue development" , "cellular response to chemical stimulus" ,"cellular response to lithium ion","clathrin-dependent endocytosis" ,"supramolecular fiber organization" ,"regulation of neuron death" ,"negative regulation of neuron death" , "spindle" , "spindle midzone"  ,	"response to external biotic stimulus", "antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway", "antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway, TAP-independent", "antigen processing and presentation of endogenous peptide antigen via MHC class Ib", 	"regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains", "regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains", "defense response",  "neuron remodeling", "neuromuscular process controlling posture", "visual behavior","response to external stimulus",	"response to biotic stimulus", "heart morphogenesis", "system process" , "skeletal muscle contraction" , "muscle system process" , "circulatory system process" , "heart process" , "muscle contraction" , "striated muscle contraction" , "heart development", "muscle organ development" , "blood circulation"  , "biological_process"   , "response to abiotic stimulus"  , "anatomical structure morphogenesis", "striated muscle tissue development" , "muscle filament sliding" , "myofibril assembly" , "cardiocyte differentiation"   , "muscle cell differentiation"   , "cellular component biogenesis"  ,"sarcomere organization"   , "cell development"  , "muscle organ morphogenesis"  , "anatomical structure formation involved in morphogenesis","tissue morphogenesis"    , "cardiac muscle tissue development" , "myosin complex"   , "myosin II complex"    ,"myofibril"     , "sarcomere"      , "Z disc"     , "A band"    , "I band" , "muscle myosin complex"     ,"striated muscle thin filament" , "circulatory system development"   , "musculoskeletal movement"         , "striated muscle cell differentiation"     , "muscle cell development"           , "striated muscle cell development"    , "cardiac myofibril assembly"     , "cardiac cell development"    , "cardiac muscle cell differentiation"   , "cardiac muscle tissue morphogenesis"  , "cardiac muscle cell development"   , "heart contraction" , "cardiac muscle contraction"  , "muscle tissue morphogenesis" , "muscle tissue development"   ,  "muscle structure development" ,"stress response to metal ion", "detoxification"   ,"spontaneous synaptic transmission"   ,  "anterograde trans-synaptic signaling"    ,"regulation of trans-synaptic signaling"    ,"synaptic signaling"    ,"trans-synaptic signaling"   , "signal release from synapse"    , "positive regulation of biomineralization"  ,           "visual system development"     "regulation of neuroinflammatory response"   , "positive regulation of neuroinflammatory response"   ,   "stress response to copper ion" ,  "eosinophil", "regulation of trans-synaptic signaling"    ,  "synaptic signaling" , "trans-synaptic signaling", "signal release from synapse"    , "positive regulation of biomineralization"  )


"stress response to metal ion", "detoxification"   ,"spontaneous synaptic transmission"   ,  "anterograde trans-synaptic signaling"    ,"regulation of trans-synaptic signaling"    ,"synaptic signaling"    ,"trans-synaptic signaling"   , "signal release from synapse"    , "positive regulation of biomineralization"  ,           "visual system development"     "regulation of neuroinflammatory response"   , "positive regulation of neuroinflammatory response"   ,   "stress response to copper ion" ,  "eosinophil", "regulation of trans-synaptic signaling"    ,  "synaptic signaling" , "trans-synaptic signaling", "signal release from synapse"    , "positive regulation of biomineralization"  

immune.related = c("neutrophil homeostasis" , "positive regulation of cytokine production" ,"adaptive immune response" , "immune effector process"  ,"immune system process" , "antigen processing and presentation of peptide antigen via MHC class Ib" , "leukocyte mediated immunity" , "dendritic cell antigen processing and presentation" , "regulation of dendritic cell antigen processing and presentation" , "regulation of immune system process", "positive regulation of immune system process", "regulation of leukocyte activation" ,"positive regulation of immune effector process", "regulation of leukocyte mediated immunity" , "positive regulation of leukocyte mediated immunity" , "regulation of adaptive immune response" , "inflammatory response" , "immune response"  , "regulation of cell death" , "prostaglandin transport" ,"regulation of icosanoid secretion" , "positive regulation of icosanoid secretion" , "regulation of prostaglandin secretion" , "positive regulation of prostaglandin secretion" , "prostaglandin secretion" , "Fc receptor signaling pathway" , "negative regulation of apoptotic process" , "regulation of immune response"     , "regulation of cell activation"   ,"negative regulation of cell death"  ,"reactive oxygen species metabolic process" ,"neutrophil clearance" ,"regulation of phospholipase C activity"  ,"L-arginine transport"   )   

gse = filter(gse, gse$Description %in% remove == FALSE)
gse.immune = filter(gse, gse$Description %in% immune.related)
gse.non.immune = filter(gse, gse$Description %in% immune.related == FALSE)

gse.immune1 = gse.immune
gse.immune1.up = filter(gse.immune1, gse.immune1$Pathway.Direction == "Upregulated")
gse.immune1.down = filter(gse.immune1, gse.immune1$Pathway.Direction == "Suppressed")

gse.immune1.up$Description = factor(gse.immune1.up$Description, levels = gse.immune1.up$Description)
gse.immune1.down$Description = factor(gse.immune1.down$Description, levels = gse.immune1.down$Description)
gse.immune1 = rbind(gse.immune1.up, gse.immune1.down)




```









# Clinical data section for DGF in Auscad
```{r, echo=FALSE, results = FALSE, warning = FALSE}
glimpse(dgf_filter)
dgf_filter$dgf_combined1
dgf_filter$donor_criteria.dcd = ifelse(dgf_filter$donor_criteria == "DCD" | 
                                         dgf_filter$donor_criteria == "DCD + ECD", 
                                       "any DCD", dgf_filter$donor_criteria )
dgf_filter$donor_criteria.ecd = ifelse(dgf_filter$donor_criteria == "DBD + ECD" | 
                                         dgf_filter$donor_criteria == "DCD + ECD", 
                                       "any ECD", dgf_filter$donor_criteria )
dgf_filter$dgf_combined1 = ifelse(dgf_filter$dgf_combined == 1, "DGF", "Control")
dgf_filter$dgf_cr_criteria1
```


# Baseline characteristics
```{r, echo=FALSE, results = FALSE, warning = FALSE}
library(gtsummary)
glimpse(dgf_filter)
colnames(dgf_filter)

baseline = c()
baseline = dgf_filter %>% select(dgf_combined1, organ_type1, HLA_mm, dsa1_pre1, dsa2_pre1,
                                 donor_criteria, donor_gender, donor_age, cit_min, dcd_wit_min,
                                 gender1, tx_age, htn_pre, dm_pre, ihd_pre, chol_pre, donor_terminal_cr,
                                 ondialysis_pretx1,renal_dx1,dgf_cr_criteria, dgf_hd_criteria,  donor_ionotropes,
                                 graft_age_months, graft_number1)
baseline.table = tbl_summary(baseline, by = dgf_combined1, missing = "no")
baseline.table1 = baseline.table %>% add_n() %>% add_p() %>% modify_header(label = "Baseline") %>% bold_labels()

types1 = c()
types1 = dgf_filter %>% select(dgf_combined1, organ_type1, donor_criteria, dgf_severity, dgf_hd_criteria, dgf_cr_criteria)
types1.table = tbl_summary(types1, by = donor_criteria, missing = "no")
types1.table = types1.table %>% add_n() %>% add_p() %>% modify_header(label = "Baseline") %>% bold_labels()

outcomes = c()
outcomes = dgf_filter %>% select(dgf_combined1, 
                                 death_functioninggraft, graft_loss, censor_followup, return_OT_01m, transfusion_01m, 
                                 cr_pre, cr_0hr,  dgf_hd_days, 
                                 cr_d1, cr_d2, cr_1m, cr_3m, cr_12m, cr_24m, 
                                 cr_36m, cr_48m, cr_60m, cr_72m, cr_84m,cr_96m, 
                                 egfr_1m, egfr_3m, egfr_12m, egfr_24m, egfr_36m, 
                                 egfr_48m, egfr_60m, egfr_72m, egfr_84m, egfr_96m,
                                 mgfr_3m, mgfr_12m, tac_d2, tac_d7, tac_d14, 
                                 tac_d21, tac_1m, tac_3m, tac_12m, 
                                 uacr_1m, uacr_3m, uacr_12m, 
                                 cmv_viremia, cmv_disease, resist_cmv, ebv_infection, 
                                 bk_viremia, bkvan, fungal_invasive_orblood, 
                                 recurrent_uti, gasteroenteritis, chest_infection, 
                                 PTLD, skin_ca, cvd_0_5y, 
                                 induct_cni, induct_apa, induct_bas, induct_extra1, 
                                 bpar_0_1m, bpar_1m, bpar_1_3m, bpar_3m, bpar_3_12m, bpar_12m, bpar_post12m,
                                 Subclin_0_1m1, subclin_1m1, subclin_1_3m1, subclin_3m1, subclin_3_12m1, subclin_12m1,
                                 bkvan_0_1m1, bkvan_1m1,bk_van_3m,  bkvan_3_12m1, bk_van_12m, bkvan_post12m1, 
                                 ci_0m, ci_1m, ci_3m, ci_12m, ct_0m, ct_1m, ct_3m, ct_12m, 
                                 iIFTA_1m, iIFTA_3m, iIFTA_12m)
outcomes.table = tbl_summary(outcomes, by = dgf_combined1, missing = "no")
outcomes.table1 = outcomes.table %>% add_n() %>% add_p() %>% modify_header(label = "Outcomes") %>% bold_labels() 

```


# Risks for DGF by regression
```{r, echo=FALSE, results = FALSE, warning = FALSE}
######
library(nnet)
model.multinominal = multinom(dgf_combined1 ~  organ_type1 + HLA_mm + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + chol_pre + 
              ondialysis_pretx1 + renal_dx1 + 
              donor_ionotropes , data = dgf_filter)

summary(model.multinominal)
exp(coef(model.multinominal))
z.multinomial = summary(model.multinominal)$coefficients/summary(model.multinominal)$standard.errors
data.frame(pnorm(abs(z.multinomial), lower.tail = FALSE)*2)

step(model.multinominal, direction = "backward")


model.multinominal1 = glm(dgf_combined1 ~  organ_type1 + HLA_mm + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + chol_pre + 
              ondialysis_pretx1 + renal_dx1 + 
              donor_ionotropes , data = dgf_filter, family = binomial)
step(model.multinominal1, direction = "backward")


model.multinominal2 = glm(dgf_combined1 ~  organ_type1 + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + chol_pre + 
              ondialysis_pretx1 + renal_dx1 + 
              donor_ionotropes , data = dgf_filter, family = binomial)
step(model.multinominal2, direction = "backward")

model.multinominal3 = glm(dgf_combined1 ~  organ_type1 + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + chol_pre + 
              ondialysis_pretx1 + 
              donor_ionotropes , data = dgf_filter, family = binomial)
step(model.multinominal3, direction = "backward")

model.multinominal4 = glm(dgf_combined1 ~  organ_type1 + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + 
              ondialysis_pretx1 + 
              donor_ionotropes , data = dgf_filter, family = binomial)
step(model.multinominal4, direction = "backward")


model.multinominal5 = glm(dgf_combined1 ~  organ_type1 + dsa1_pre1 + dsa2_pre1 +
              donor_criteria + donor_gender + 
              gender1 +  htn_pre + dm_pre + ihd_pre + 
              ondialysis_pretx1  , data = dgf_filter, family = binomial)
step(model.multinominal5, direction = "backward")


model.multinominal6 = glm(dgf_combined1 ~   donor_criteria +  ondialysis_pretx1  , data = dgf_filter, family = binomial)
step(model.multinominal6, direction = "backward")
summary(model.multinominal6)
exp(coef(model.multinominal6))
```



```{r, echo=FALSE, results = FALSE, warning = FALSE}
#######
model.linear = glm(dgf_combined1 ~ donor_age + cit_min + dcd_wit_min + tx_age + donor_terminal_cr +
              graft_number1, data = dgf_filter, family = binomial)
summary(model.linear)
step(model.linear, direction = "backward")

model.linear1 = glm(dgf_combined1 ~ donor_age + cit_min + 
                     donor_terminal_cr + dcd_wit_min + 
                      graft_number1, data = dgf_filter, family = binomial)
step(model.linear1, direction = "backward")
summary(model.linear1)

model.linear2 = glm(dgf_combined1 ~ donor_age + cit_min + 
                     donor_terminal_cr + dcd_wit_min, 
                    data = dgf_filter, family = binomial)
step(model.linear2, direction = "backward")
summary(model.linear2)

model.linear3 = glm(dgf_combined1 ~ donor_age + 
                     donor_terminal_cr + dcd_wit_min, 
                    data = dgf_filter, family = binomial)
step(model.linear3, direction = "backward")
summary(model.linear3)

model.linear4 = glm(dgf_combined1 ~ donor_age + 
                     donor_terminal_cr,
                    data = dgf_filter, family = binomial)
step(model.linear4, direction = "backward")
summary(model.linear4)

model.linear5 = glm(dgf_combined1 ~
                     donor_terminal_cr,
                    data = dgf_filter, family = binomial)
step(model.linear5, direction = "backward")
summary(model.linear5)
exp(coef(model.linear5))
# OR 1.009 (p = 0.000206 )
```


# Survival curves
```{r, echo=FALSE, results = FALSE, warning = FALSE}

HR.table = coxph(Surv(dgf_filter$bpar_any) ~ 
                     dgf_filter$donor_criteria + dgf_filter$dgf_combined1 + dgf_filter$dgf_severity, 
                   data = dgf_filter) %>% tbl_regression(exponentiate = TRUE)

HR.table1 = coxph(Surv(dgf_filter$subclin_any) ~ 
                     dgf_filter$donor_criteria + dgf_filter$dgf_combined1+ dgf_filter$dgf_severity, 
                   data = dgf_filter) %>% tbl_regression(exponentiate = TRUE)

HR.table2 = coxph(Surv(dgf_filter$graft_loss) ~ 
                     dgf_filter$donor_criteria + dgf_filter$dgf_criteria+ dgf_filter$dgf_severity, 
                   data = dgf_filter) %>% tbl_regression(exponentiate = TRUE)

HR.table3 = coxph(Surv(ifelse(dgf_filter$graft_loss ==1 | dgf_filter$death == 1, 1,0 )) ~ 
                     dgf_filter$donor_criteria + dgf_filter$dgf_combined1+ dgf_filter$dgf_severity, 
                   data = dgf_filter) %>% tbl_regression(exponentiate = TRUE)

```

# Basic plots
```{r, echo=FALSE, results = FALSE, warning = FALSE}
ggplot(dgf_filter, aes(y = dgf_filter$mgfr_12m, colour = bpar_any1)) + geom_boxplot() + facet_grid(~ dgf_combined1) + theme_bw()
ggplot(dgf_filter, aes(y = dgf_filter$cadi_12m, colour = dgf_filter$bpar_0_1m1)) + geom_boxplot() + facet_grid(~ dgf_combined1) + theme_bw()
ggplot(dgf_filter, aes(y = dgf_filter$cadi_12m, colour = dgf_filter$bpar_1m1)) + geom_boxplot() + facet_grid(~ dgf_combined1) + theme_bw()
ggplot(dgf_filter, aes(y = cadi_12m, fill = dgf_combined1, x = donor_criteria )) + 
  geom_boxplot() + facet_grid(bpar_any1 ~.) + theme_bw() + scale_fill_brewer(palette="Set2")
ggplot(dgf_filter, aes(y = mgfr_12m, fill = dgf_combined1, x = donor_criteria )) + 
  geom_boxplot() + facet_grid(bpar_any1 ~.) + theme_bw() + scale_fill_brewer(palette="Set2")
   # simple plots 
    tage<-ggplot(dgf_filter, aes(x=dgf_criteria, y=tx_age, fill=dgf_criteria)) +
      geom_boxplot(alpha=0.3) +theme_bw()+
      theme(legend.position="none")+ 
      ggtitle("DGF vs Recipient Age")+
      xlab("Groups")+
      ylab("Recipient age (years)")
    
    dage<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=donor_age, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +theme_bw()+
      theme(legend.position="none")+
      ggtitle("DGF vs Donor age")+
      xlab("Groups")+
      ylab("Donor age (years)")

    cit1<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cit_min, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +theme_bw()+
      theme(legend.position="none")+
      ggtitle("DGF vs Cold Ischemic Time")+
      xlab("Groups")+
      ylab("CIT (mins)")
    
    swit1<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=anastamosis_time_mins, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +theme_bw()+
      theme(legend.position="none")+
      ggtitle("DGF vs Anastamosis time")+
      xlab("Groups")+
      ylab("Anastamosis time (mins)")
    
    durea<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=donor_terminal_urea, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme(legend.position="none")+theme_bw()+
      ggtitle("DGF vs Terminal Ur")+
      xlab("Groups")+
      ylab("Donor Urea")
        
    dcr<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=donor_terminal_cr, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +theme_bw()+
      theme(legend.position="none")+
      ggtitle("DGF vs Terminal SCr")+
      xlab("Groups")+
      ylab("Donor SCr")
    
    ggarrange(tage, dage, cit1, swit1, fontsize = 0.5, ncol=2, nrow=2)
    
    #ggplot(dgf_filter, aes(x=renal_dx)) +
    #  geom_histogram(color="darkblue", fill="lightblue")+
    #  xlab("Renal Diagnosis")
    
    ggplot(dgf_filter, aes(fill=renal_dx1, x=dgf_criteria1)) + 
      geom_bar(position="dodge", colour = "black")+ theme_bw()+
      xlab("Renal Diagnosis")+      
      ggtitle("Renal Diagnosis")
    
    ggplot(dgf_filter, aes(fill=donor_criteria_combined, x=dgf_criteria1)) + 
      geom_bar(position="dodge", colour = "black")+ theme_bw()+
      xlab("Donor Criteria")+
      ggtitle("Donor Criteria")
    
    base1<-ggplot(dgf_filter, aes(fill=donor_criteria_base1, x=dgf_criteria1)) + 
      geom_bar(position="dodge", colour = "black")+ theme_bw()+
      xlab("Donor Criteria")+
      ggtitle("Donor Criteria")
    
    base2<- ggplot(dgf_filter, aes(fill=donor_criteria_ECD1, x=dgf_criteria1)) + 
      geom_bar(position="dodge", colour = "black")+ theme_bw()+
      xlab("Donor Criteria")+
      ggtitle("Donor Criteria - ECD")
    
    ggarrange(base1,base2)
```



```{r, echo=FALSE, results = FALSE, warning = FALSE}
########################
# SURVIVAL CURVES
########################
# select the correct legend.lab section for each

sp1 = ggsurvplot(survfit(Surv(time_to_death_old1, death)
                   ~ dgf_criteria,   data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0, 1), 
          title = "Death",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "bottom",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 

sp2 =ggsurvplot(survfit(Surv(time_to_graft_loss1, graft_loss)
                   ~ dgf_criteria,   data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0.8, 1), 
          title = "Graft loss",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "bottom",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 

sp2a =ggsurvplot(survfit(Surv(time_to_graft_loss1, graft_loss)
                   ~ donor_criteria,   data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0, 1), 
          title = "Graft loss",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "right",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5)

sp3 =ggsurvplot(survfit(Surv(time_to_bpar1, bpar_any)
                   ~ dgf_criteria,  data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,12),
          ylim = c(0, 1), 
          title = "Biopsy proven rejection",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "bottom",
          break.x.by=3,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 


sp4 =ggsurvplot(survfit(Surv(time_to_bpar1, bpar_any)
                   ~ dgf_severity,  data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,12),
          ylim = c(0, 1), 
          title = "Biopsy proven rejection",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "right",
          break.x.by=3,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 

sp5 = ggsurvplot(survfit(Surv(time_to_subclin1, subclin_any)
                   ~ dgf_criteria,  data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0, 1), 
          title = "Subclinical rejection",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "bottom",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 

sp6 = ggsurvplot(survfit(Surv(time_to_subclin1, subclin_any)
                   ~ dgf_severity,  data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE, 
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0, 1), 
          title = "Subclinical rejection",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "right",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5) 

ggsurvplot(survfit(Surv(time_to_subclin1, subclin_any)
                   ~ donor_criteria,  data=dgf_filter[dgf_filter$graft_age_months >=12,]),
          pval = TRUE,
          conf.int = FALSE,
          # risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          # surv.median.line = "hv", # Specify median survival
          ylab = "Event-free survival",
          xlab = "Time (months)",
          xlim = c(0,60),
          ylim = c(0, 1),
          title = "Subclin",
          # legend.labs = c("Control", "DGF"),
          #legend.labs = c("Control", "<20%drop", "HD<7 days", "HD>7days"),
          legend = "right",
          break.x.by=6,
          risk.table = FALSE, # Add risk table
          # risk.table.height = 0.3,
          # tables.theme = clean_table_theme(),
          # ggtheme = theme_classic(),
          fontsize=5)


coxph(Surv(time_to_subclin1, subclin_any)
                   ~ dgf_severity,  data=dgf_filter[dgf_filter$graft_age_months >=12,])
# jpeg("sp2.jpeg", width = 600, height = 400)
# sp2
# dev.off()

sp6

```



```{r}
######################
# COX PROPROTIONAL HAZARDS for rejection
######################
# Univariate analysis by Cox PH model
#coxph(Surv(dgf_filter$time_to_graft_loss, dgf_filter$graft_loss_kidneyonly) ~ dgf_filter$dgf_criteria1, data = dgf_filter)
# summary(res.cox)

# assumptions
dgf_filter$graft_number <- replace_na(dgf_filter$bpar_any, 1)
dgf_filter$bpar_any2 <- replace_na(dgf_filter$bpar_any, 0)
dgf_filter$graft_number2 <- ifelse(dgf_filter$graft_number>0,1, 0)
dgf_filter$induct_extra2 <- ifelse(dgf_filter$induct_extra>0, 1, 0)

# to run multiple variables into the univariate model
covariates <- c("dgf_criteria", "Subclin_0_1m", "dcd", "dbd", "ecd", "living", "dcdecd","graft_number2", "induct_extra2", "HLA_mm1", "dsa1_pre", "dsa2_pre", "donor_age", "tx_age", "cit_min", "anastamosis_time_mins", "diabetickidney")

# DEATH
univ_formulas.death <- sapply(covariates,
                        function(x) as.formula(paste('Surv(dgf_filter$time_to_death_old, 
                                                     dgf_filter$death)~', x)))
univ_models.death <- lapply(univ_formulas.death, function(x){coxph(x, data = dgf_filter)})

# Extract data 
univ_results.death <- lapply(univ_models.death,
                       function(x){ 
                          x <- summary(x)
                          p.value<-signif(x$wald["pvalue"], digits=2)
                          wald.test<-signif(x$wald["test"], digits=2)
                          beta<-signif(x$coef[1], digits=2);#coeficient beta
                          HR <-signif(x$coef[2], digits=2);#exp(beta)
                          HR.confint.lower <- signif(x$conf.int[,"lower .95"], 2)
                          HR.confint.upper <- signif(x$conf.int[,"upper .95"],2)
                          HR <- paste0(HR, " (", 
                                       HR.confint.lower, "-", HR.confint.upper, ")")
                          res<-c(beta, HR, wald.test, p.value)
                          names(res)<-c("beta", "HR (95% CI for HR)", "wald.test", 
                                        "p.value")
                          return(res)
                          #return(exp(cbind(coef(x),confint(x))))
                         })
res.death <- t(as.data.frame(univ_results.death, check.names = FALSE))
as.data.frame(res.death)
res.death<-res.death[,c(1:4)]
colnames(res.death)<- c("beta", "HR (95% CI for HR)", "wald.test", "p.value")
htmlTable((res.death))

# multivariate version
res.cox.death <- coxph(Surv(dgf_filter$time_to_death_old, dgf_filter$death) ~ dgf_criteria, data= dgf_filter)
summary(res.cox.death)
test.ph <- cox.zph(res.cox.death)
test.ph

# GRAFT LOSS (KIDNEY)
univ_formulas.loss <- sapply(covariates,
                        function(x) as.formula(paste('Surv(dgf_filter$time_to_graft_loss, 
                                                     dgf_filter$graft_loss_kidneyonly)~', x)))
univ_models.loss <- lapply(univ_formulas.loss, function(x){coxph(x, data = dgf_filter)})

# Extract data 
univ_results.loss <- lapply(univ_models.loss,
                       function(x){ 
                          x <- summary(x)
                          p.value<-signif(x$wald["pvalue"], digits=2)
                          wald.test<-signif(x$wald["test"], digits=2)
                          beta<-signif(x$coef[1], digits=2);#coeficient beta
                          HR <-signif(x$coef[2], digits=2);#exp(beta)
                          HR.confint.lower <- signif(x$conf.int[,"lower .95"], 2)
                          HR.confint.upper <- signif(x$conf.int[,"upper .95"],2)
                          HR <- paste0(HR, " (", 
                                       HR.confint.lower, "-", HR.confint.upper, ")")
                          res<-c(beta, HR, wald.test, p.value)
                          names(res)<-c("beta", "HR (95% CI for HR)", "wald.test", 
                                        "p.value")
                          return(res)
                          #return(exp(cbind(coef(x),confint(x))))
                         })
res.loss <- t(as.data.frame(univ_results.loss, check.names = FALSE))
as.data.frame(res.loss)
res.loss<-res.loss[,c(1:4)]
colnames(res.loss)<- c("beta", "HR (95% CI for HR)", "wald.test", "p.value")
htmlTable((res.loss))

# multivariate version
res.cox.loss <- coxph(Surv(dgf_filter$time_to_graft_loss, dgf_filter$graft_loss_kidneyonly) ~ dgf_criteria1 , data =  dgf_filter)
summary(res.cox.loss)


# BPAR DGF vs control
univ_formulas.reject <- sapply(covariates,
                        function(x) as.formula(paste('Surv(dgf_filter$time_to_subclin, 
                                                     dgf_filter$subclin_any)~', x)))
univ_models.reject<- lapply(univ_formulas.reject, function(x){coxph(x, data = dgf_filter)})
univ_models.reject
# Extract data 
univ_results.reject <- lapply(univ_models.reject,
                       function(x){ 
                          x <- summary(x)
                          p.value<-signif(x$wald["pvalue"], digits=2)
                          wald.test<-signif(x$wald["test"], digits=2)
                          beta<-signif(x$coef[1], digits=2);#coeficient beta
                          HR <-signif(x$coef[2], digits=2);#exp(beta)
                          HR.confint.lower <- signif(x$conf.int[,"lower .95"], 2)
                          HR.confint.upper <- signif(x$conf.int[,"upper .95"],2)
                          HR <- paste0(HR, " (", 
                                       HR.confint.lower, "-", HR.confint.upper, ")")
                          res<-c(beta, HR, wald.test, p.value)
                          names(res)<-c("beta", "HR (95% CI for HR)", "wald.test", 
                                        "p.value")
                          return(res)
                          #return(exp(cbind(coef(x),confint(x))))
                         })
res.reject <- t(as.data.frame(univ_results.reject, check.names = FALSE))
as.data.frame(res.reject)
res.reject<-res.reject[,c(1:4)]
colnames(res.reject)<- c("beta", "HR (95% CI for HR)", "wald.test", "p.value")
htmlTable((res.reject))
univ_results.reject

# multivariate version
res.cox.reject <- coxph(Surv(dgf_filter$time_to_bpar,dgf_filter$bpar_any2) ~ dgf_criteria1+ dsa2_pre, data = dgf_filter)
summary(res.cox.reject)
test.ph <- cox.zph(res.cox.reject)
test.ph

###
# BPAR DGF > 7 days vs no DGF
###

dgf7<-subset(dgf_filter, dgf_filter$dgf_hd_days<1 | dgf_filter$dgf_hd_days>6)
dgf7$hd <-ifelse(dgf7$dgf_hd_days>=7,1,0)
dgf7$subclinto3m <-ifelse(dgf7$Subclin_0_1m==1|dgf7$subclin_1m==1|dgf7$subclin_3m==1,1,0)

dgf7$ecd <- ifelse(dgf7$donor_criteria_ECD ==1,1,0)
dgf7$dcdecd <-ifelse(dgf7$donor_criteria_base ==1 & dgf7$donor_criteria_ECD==1,1,0)
dgf7$dcd <-ifelse(dgf7$donor_criteria_base ==1 ,1,0)
dgf7$dbd <-ifelse(dgf7$donor_criteria_base == 0,1,0)
dgf7$living<-ifelse(dgf7$donor_criteria_base ==3,1,0)

# other baseline variables
dgf7$diabetickidney <- ifelse(dgf7$renal_dx==2,1,0)
dgf7$ktxall<-ifelse(dgf7$organ_type !=3 ,1,0)
dgf7$ktxcad<-ifelse(dgf7$organ_type ==1 ,1,0)
dgf7$graft_number2 <-ifelse(dgf7$graft_number>1,1,0)
dgf7$induct_extra2 <-ifelse(dgf7$induct_extra >0,1,0)

covariates1 <- c("hd", "Subclin_0_1m", "dcd", "dbd", "ecd", "living", "dcdecd","graft_number2", "induct_extra2", "HLA_mm", "dsa1_pre", "dsa2_pre", "donor_age", "tx_age", "cit_min", "anastamosis_time_mins", "diabetickidney")

# subclinical rejection
univ_formulas.reject <- sapply(covariates1,
                        function(x) as.formula(paste('Surv(dgf7$time_to_subclin, 
                                                     dgf7$subclin_any)~', x)))
univ_models.reject<- lapply(univ_formulas.reject, function(x){coxph(x, data = dgf7)})

# Extract data 
univ_results.reject <- lapply(univ_models.reject,
                       function(x){ 
                          x <- summary(x)
                          p.value<-signif(x$wald["pvalue"], digits=2)
                          wald.test<-signif(x$wald["test"], digits=2)
                          beta<-signif(x$coef[1], digits=2);#coeficient beta
                          HR <-signif(x$coef[2], digits=2);#exp(beta)
                          HR.confint.lower <- signif(x$conf.int[,"lower .95"], 2)
                          HR.confint.upper <- signif(x$conf.int[,"upper .95"],2)
                          HR <- paste0(HR, " (", 
                                       HR.confint.lower, "-", HR.confint.upper, ")")
                          res<-c(beta, HR, wald.test, p.value)
                          names(res)<-c("beta", "HR (95% CI for HR)", "wald.test", 
                                        "p.value")
                          return(res)
                          #return(exp(cbind(coef(x),confint(x))))
                         })
res.reject <- t(as.data.frame(univ_results.reject, check.names = FALSE))
as.data.frame(res.reject)
res.reject<-res.reject[,c(1:4)]
colnames(res.reject)<- c("beta", "HR (95% CI for HR)", "wald.test", "p.value")
htmlTable((res.reject))
univ_results.reject

# multivariate version
res.cox.reject <- coxph(Surv(dgf7$time_to_subclin,dgf7$subclin_any) ~  hd , data = dgf7)
summary(res.cox.reject)
test.ph <- cox.zph(res.cox.reject)
test.ph


```


```{r}
##############################
# Outcome data by DGF (graft function)
##############################

# descriptive table
table1(~dgf_filter$cr_1m + dgf_filter$egfr_1m + dgf_filter$uacr_1m +dgf_filter$cr_3m +dgf_filter$egfr_3m +dgf_filter$uacr_3m +dgf_filter$mgfr_3m +dgf_filter$cr_12m +dgf_filter$egfr_12m +dgf_filter$uacr_12m +dgf_filter$mgfr_12m +dgf_filter$cr_24m +dgf_filter$egfr_24m +dgf_filter$cr_36m +dgf_filter$egfr_36m +dgf_filter$cr_48m +dgf_filter$egfr_48m +dgf_filter$cr_60m +dgf_filter$egfr_60m +dgf_filter$mgfr_36m +dgf_filter$mgfr_60m| dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
cr1m_dgf<- kruskal.test(dgf_filter$cr_1m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr3m_dgf<- kruskal.test(dgf_filter$cr_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr12m_dgf<- kruskal.test(dgf_filter$cr_12m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr24m_dgf<- kruskal.test(dgf_filter$cr_24m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr36m_dgf<- kruskal.test(dgf_filter$cr_36m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr48m_dgf<- kruskal.test(dgf_filter$cr_48m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cr60m_dgf<- kruskal.test(dgf_filter$cr_60m ~ dgf_filter$dgf_criteria, data = dgf_filter)

egfr1m_dgf<- kruskal.test(dgf_filter$egfr_1m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr3m_dgf<- kruskal.test(dgf_filter$egfr_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr12m_dgf<- kruskal.test(dgf_filter$egfr_12m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr24m_dgf<- kruskal.test(dgf_filter$egfr_24m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr36m_dgf<- kruskal.test(dgf_filter$egfr_36m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr48m_dgf<- kruskal.test(dgf_filter$egfr_48m ~ dgf_filter$dgf_criteria, data = dgf_filter)
egfr60m_dgf<- kruskal.test(dgf_filter$egfr_60m ~ dgf_filter$dgf_criteria, data = dgf_filter)

uacr1m_dgf<- kruskal.test(dgf_filter$uacr_1m ~ dgf_filter$dgf_criteria, data = dgf_filter)
uacr3m_dgf<- kruskal.test(dgf_filter$uacr_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)
uacr12m_dgf<- kruskal.test(dgf_filter$uacr_12m ~ dgf_filter$dgf_criteria, data = dgf_filter)

mgfr3m_dgf<- kruskal.test(dgf_filter$mgfr_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)
mgfr12m_dgf<- kruskal.test(dgf_filter$mgfr_12m ~ dgf_filter$dgf_criteria, data = dgf_filter)
mgfr36m_dgf<- kruskal.test(dgf_filter$mgfr_36m ~ dgf_filter$dgf_criteria, data = dgf_filter)
mgfr60m_dgf<- kruskal.test(dgf_filter$mgfr_60m ~ dgf_filter$dgf_criteria, data = dgf_filter)

# summarise the p.values for graft bchem
  p.graft.bchm.var <-c("Cr (1m)", "Cr (3m)", "Cr (12m)", "Cr (24m)", "Cr (36m)", "Cr (48m)", "Cr (60m)",
                       "eGFR (1m)", "eGFR (3m)", "eGFR (12m)", "eGFR (24m)", "eGFR (36m)", "eGFR (48m)", "eGFR (60m)", 
                       "uACR (1m)", "uACR (3m)", "uACR (12m)", "mGFR (3m)", "mGFR (12m)", "mGFR (36m)", "mGFR (60)")
                       

    # put together stats
  p.graft.bchm.dgf<-c(cr1m_dgf$p.value, cr3m_dgf$p.value, cr12m_dgf$p.value, cr24m_dgf$p.value, cr36m_dgf$p.value, 
                      cr48m_dgf$p.value, cr60m_dgf$p.value, egfr1m_dgf$p.value, egfr3m_dgf$p.value, egfr12m_dgf$p.value, 
                      egfr24m_dgf$p.value, egfr36m_dgf$p.value, egfr48m_dgf$p.value, egfr60m_dgf$p.value, uacr1m_dgf$p.value, 
                      uacr3m_dgf$p.value, uacr12m_dgf$p.value, mgfr3m_dgf$p.value, mgfr12m_dgf$p.value, mgfr36m_dgf$p.value, 
                      mgfr60m_dgf$p.value)
    
  p.graft.bchm.dgf<-round(p.graft.bchm.dgf, digits = 3)
  p.graft.bchm1 <- cbind(p.graft.bchm.var, p.graft.bchm.dgf)
    htmlTable((p.graft.bchm1))
```


```{r}
##################
# DGF > 7 days
##################

####### Outcome data by DGF (graft function)
dgf7<-subset(dgf_filter, dgf_filter$dgf_hd_days<1 | dgf_filter$dgf_hd_days>6)
dgf7$hd <-ifelse(dgf7$dgf_hd_days>=7,1,0)
dgf7$bparto3m <-ifelse(dgf7$Subclin_0_1m==1|dgf7$subclin_1m==1|dgf7$subclin_3m==1,1,0)

# descriptive table
table1(~dgf7$cr_1m + dgf7$egfr_1m + dgf7$uacr_1m +dgf7$cr_3m +dgf7$egfr_3m +dgf7$uacr_3m +dgf7$mgfr_3m +dgf7$cr_12m +dgf7$egfr_12m +dgf7$uacr_12m +dgf7$mgfr_12m +dgf7$cr_24m +dgf7$egfr_24m +dgf7$cr_36m +dgf7$egfr_36m +dgf7$cr_48m +dgf7$egfr_48m +dgf7$cr_60m +dgf7$egfr_60m +dgf7$mgfr_36m +dgf7$mgfr_60m| dgf7$hd, data=dgf7, overall="Total", render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
cr1m_dgf<- kruskal.test(dgf7$cr_1m ~ dgf7$hd, data = dgf_filter)
cr3m_dgf<- kruskal.test(dgf7$cr_3m ~ dgf7$hd, data = dgf_filter)
cr12m_dgf<- kruskal.test(dgf7$cr_12m ~ dgf7$hd, data = dgf_filter)
cr24m_dgf<- kruskal.test(dgf7$cr_24m ~ dgf7$hd, data = dgf_filter)
cr36m_dgf<- kruskal.test(dgf7$cr_36m ~ dgf7$hd, data = dgf_filter)
cr48m_dgf<- kruskal.test(dgf7$cr_48m ~ dgf7$hd, data = dgf_filter)
cr60m_dgf<- kruskal.test(dgf7$cr_60m ~ dgf7$hd, data = dgf_filter)

egfr1m_dgf<- kruskal.test(dgf7$egfr_1m ~ dgf7$hd, data = dgf_filter)
egfr3m_dgf<- kruskal.test(dgf7$egfr_3m ~ dgf7$hd, data = dgf_filter)
egfr12m_dgf<- kruskal.test(dgf7$egfr_12m ~ dgf7$hd, data = dgf_filter)
egfr24m_dgf<- kruskal.test(dgf7$egfr_24m ~ dgf7$hd, data = dgf_filter)
egfr36m_dgf<- kruskal.test(dgf7$egfr_36m ~ dgf7$hd, data = dgf_filter)
egfr48m_dgf<- kruskal.test(dgf7$egfr_48m ~ dgf7$hd, data = dgf_filter)
egfr60m_dgf<- kruskal.test(dgf7$egfr_60m ~ dgf7$hd, data = dgf_filter)

mgfr3m_dgf<- kruskal.test(dgf7$mgfr_3m ~ dgf7$hd, data = dgf_filter)
mgfr12m_dgf<- kruskal.test(dgf7$mgfr_12m ~ dgf7$hd, data = dgf_filter)
mgfr36m_dgf<- kruskal.test(dgf7$mgfr_36m ~ dgf7$hd, data = dgf_filter)
mgfr60m_dgf<- kruskal.test(dgf7$mgfr_60m ~ dgf7$hd, data = dgf_filter)

# summarise the p.values for graft bchem
  p.graft.bchm.var <-c("Cr (1m)", "Cr (3m)", "Cr (12m)", "Cr (24m)", "Cr (36m)", "Cr (48m)", "Cr (60m)",
                       "eGFR (1m)", "eGFR (3m)", "eGFR (12m)", "eGFR (24m)", "eGFR (36m)", "eGFR (48m)", "eGFR (60m)", 
                       "mGFR (3m)", "mGFR (12m)", "mGFR (36m)", "mGFR (60)")
                       

    # put together stats
  p.graft.bchm.dgf<-c(cr1m_dgf$p.value, cr3m_dgf$p.value, cr12m_dgf$p.value, cr24m_dgf$p.value, cr36m_dgf$p.value, 
                      cr48m_dgf$p.value, cr60m_dgf$p.value, egfr1m_dgf$p.value, egfr3m_dgf$p.value, egfr12m_dgf$p.value, 
                      egfr24m_dgf$p.value, egfr36m_dgf$p.value, egfr48m_dgf$p.value, egfr60m_dgf$p.value, mgfr3m_dgf$p.value, mgfr12m_dgf$p.value, 
                      mgfr36m_dgf$p.value, mgfr60m_dgf$p.value)
    
  p.graft.bchm.dgf<-round(p.graft.bchm.dgf, digits = 3)
  p.graft.bchm1 <- cbind(p.graft.bchm.var, p.graft.bchm.dgf)
    htmlTable((p.graft.bchm1))
    
     # bpar for dgf7
     table1(~dgf7$bpar_any1 + dgf7$subclin_any1 + dgf7$bkvan_any1 +dgf7$mpred_any1 +dgf7$steroidres_any1
           |dgf7$hd, data=dgf_filter,overall="Total", 
           render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))
    
    bpar_anydgf<- chisq.test(dgf7$bpar_any, dgf7$hd, correct = FALSE)
    subclin_anydgf<- chisq.test(dgf7$subclin_any, dgf7$hd, correct = FALSE)
    bkvan_anydgf<- chisq.test(dgf7$bkvan_any, dgf7$hd, correct = FALSE)
    mpred_anydgf<- chisq.test(dgf7$mpred_any, dgf7$hd, correct = FALSE)
    steroidres_anydgf<- chisq.test(dgf7$steroidres_any, dgf7$hd, correct = FALSE)
    
    bpar_anydgf$p.value
    subclin_anydgf$p.value
    bkvan_anydgf$p.value
    mpred_anydgf$p.value
    steroidres_anydgf$p.value
    
    p.graft.reject_any.var <-c("BPAR", "Subclin", "BKVAN", "Mpred", "Additional IS")
    p.graft.reject_any.dgf<-c(bpar_anydgf$p.value, subclin_anydgf$p.value, bkvan_anydgf$p.value, mpred_anydgf$p.value,
                              steroidres_anydgf$p.value)
    p.graft.reject_any.dgf<-round(p.graft.reject_any.dgf, digits = 3)
    p.graft.reject2 <- cbind(p.graft.reject_any.var, p.graft.reject_any.dgf)
    htmlTable((p.graft.reject2))
    
    # descriptive table
table1(~dgf7$cadi_1m + dgf7$cadi_3m + dgf7$cadi_12m| dgf7$dgf_criteria1, data=dgf7, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
cadi1m_dgf<- kruskal.test(dgf7$cadi_1m ~ dgf7$hd, data = dgf7)
cadi3m_dgf<- kruskal.test(dgf7$cadi_3m ~ dgf7$hd, data = dgf7)
cadi12m_dgf<- kruskal.test(dgf7$cadi_12m ~ dgf7$hd, data = dgf7)

# summarise the p.values for graft bchem
  p.graft.cadi.var <-c("CADI (1m)","CADI(3m))", "CADI (12m)")
                       
  p.graft.cadi.dgf<-c(cadi1m_dgf$p.value, cadi3m_dgf$p.value, cadi12m_dgf$p.value)
    
  p.graft.cadi.dgf<-round(p.graft.cadi.dgf, digits = 3)
  p.graft.cadi1 <- cbind(p.graft.cadi.var, p.graft.cadi.dgf)
    htmlTable((p.graft.cadi1))
    
# CADI scores stratified by level = 2
    dgf7$cadi1m_2 <- ifelse(dgf7$cadi_1m >=2, 1,0)
    dgf7$cadi1m_2 <- as.factor(dgf7$cadi1m_2)
    dgf7$cadi3m_2 <- ifelse(dgf7$cadi_3m >=2, 1,0)
    dgf7$cadi3m_2 <- as.factor(dgf7$cadi3m_2)
    dgf7$cadi12m_2 <- ifelse(dgf7$cadi_12m >=2, 1,0)
    dgf7$cadi12m_2 <- as.factor(dgf7$cadi12m_2)
    
    table1(~dgf7$cadi1m_2 + dgf7$cadi3m_2 + dgf7$cadi12m_2| dgf7$hd, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))
    
    cadi1m2_dgf<- chisq.test(dgf7$cadi1m_2, dgf7$hd, correct = FALSE)
    cadi3m2_dgf<- chisq.test(dgf7$cadi3m_2, dgf7$hd, correct = FALSE)
    cadi12m2_dgf<- chisq.test(dgf7$cadi12m_2, dgf7$hd, correct = FALSE)

# summarise the p.values for graft bchem
  p.graft.cadi.var <-c("CADI (1m)","CADI(3m))", "CADI (12m)")
                       
  p.graft.cadi.dgf<-c(cadi1m2_dgf$p.value, cadi3m2_dgf$p.value, cadi12m2_dgf$p.value)
    
  p.graft.cadi.dgf<-round(p.graft.cadi.dgf, digits = 3)
  p.graft.cadi2 <- cbind(p.graft.cadi.var, p.graft.cadi.dgf)
    htmlTable((p.graft.cadi2))
```


```{r}
##############
# simple plots for diff groups if strata different
################
    pcr1m<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=cr_1m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10) + theme(legend.position="none")+ 
      xlab("Groups")+ ylab("SCr (umol/L)") + ggtitle("SCr (1m)")

    pegfr1m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=egfr_1m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups")+ ylab("eGFR (ml/min/1.73m2)") + ggtitle("eGFR (1m)")
    
    pcr3m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_3m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups") + ggtitle("SCr (3m)") + ylab("SCr (umol/L)")
    
    pegfr3m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=egfr_3m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups")+ ggtitle("eGFR (3m)")+ ylab("eGFR (ml/min/1.73m2)")
        
    pcr12m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_12m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups")+ ggtitle("SCr (12m)")+ ylab("SCr (umol/L)")
    
    pegfr12m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=egfr_12m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups")+ ggtitle("eGFR (12m)")+ ylab("eGFR (ml/min/1.73m2)")
    
     pcr24m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_24m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("SCr (24m)")
          
     pcr36m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_36m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("SCr (36m)")
               
     pcr48m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_48m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("SCr (48m)")
                    
     pcr60m<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=cr_60m, fill=dgf_criteria1)) +
      geom_boxplot(alpha=0.3) +
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("SCr (60m)")
     
    pmgfr3m<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=mgfr_3m, fill=donor_criteria_base1)) +
      geom_boxplot(alpha=0.3) +
      theme_bw(base_size=10)+theme(legend.position="none")+
      xlab("Groups")+ ylab("mGFR (ml/min/1.73m2)") + ggtitle("mGFR (3m)")
         
    pmgfr12m<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=mgfr_12m, fill=donor_criteria_base1)) +
      geom_boxplot(alpha=0.3) +
     theme_bw(base_size=10)+ theme(legend.position="none")+ 
    xlab("Groups")+ ylab("mGFR (ml/min/1.73m2)") + ggtitle("mGFR (12m)")
             
    ggarrange(pcr1m, pcr3m, pcr12m, pmgfr3m,pegfr1m,pegfr3m, pegfr12m, pmgfr12m,ncol=4, nrow= 2)
```


```{r}

#############
#more plots of Cr
#############

plotcr <- dgf_filter[,c("cr_1m", "cr_3m", "cr_12m", "cr_24m", "cr_36m", "cr_48m", "cr_60m", "dgf_criteria")]
plotcr<-melt(plotcr, id=c("dgf_criteria"))
colnames(plotcr)<-c("dgf_criteria", "time", "creatinine")
plotcr$dgf_criteria<-as.factor(plotcr$dgf_criteria)

plotegfr <- dgf_filter[,c("egfr_1m", "egfr_3m", "egfr_12m", "egfr_24m", "egfr_36m", "egfr_48m", "egfr_60m", "dgf_criteria")]
plotegfr<-melt(plotegfr, id=c("dgf_criteria"))
colnames(plotegfr)<-c("dgf_criteria", "time", "eGFR")
plotegfr$dgf_criteria<-as.factor(plotegfr$dgf_criteria)

plotmgfr<- dgf_filter[,c("mgfr_3m", "mgfr_12m", "dgf_criteria")]
plotmgfr<-melt(plotmgfr, id=c("dgf_criteria"))
colnames(plotmgfr)<-c("dgf_criteria", "time", "mGFR")
plotmgfr$dgf_criteria<-as.factor(plotmgfr$dgf_criteria)

cr1a<-ggplot(plotcr, aes(x=time, y=creatinine, fill=dgf_criteria)) + 
  geom_boxplot() +
  theme_bw()+
  xlab("Time point")+
  ylab("Creatinine (umol/L)")

cr2a<-ggplot(plotegfr, aes(x=time, y=eGFR, fill=dgf_criteria)) + 
  geom_boxplot() + theme_bw()+
  xlab("Time point")+
  ylab("eGFR (ml/min/1.73m2))")

cr3a<-ggplot(plotmgfr, aes(x=time, y=mGFR, fill=dgf_criteria)) + 
  geom_boxplot() + theme_bw()+
  xlab("Time point")+
  ylab("mGFR (ml/min/1.73m2))")

ggplot(plotmgfr, aes(x=time, y=mGFR, fill=dgf_criteria)) + 
  geom_boxplot() + theme_bw()+
  xlab("Time point")+
  ylab("mGFR (ml/min/1.73m2))")

ggarrange(cr1a, cr2a, ncol=1, rcol=2)

```


```{r}
#########################
# Outcome data by DGF (rejection)
#########################

# descriptive table
table1(~dgf_filter$bpar_0_1m1 + dgf_filter$Subclin_0_1m1 +dgf_filter$mpred_0_1m1 +dgf_filter$steroid_res_0_1m1 + dgf_filter$bkvan_0_1m1 
       +dgf_filter$bpar_1m1 + dgf_filter$subclin_1m1 +dgf_filter$mpred_1m1 +dgf_filter$steroid_res_1m1 + dgf_filter$bkvan_1m1
       +dgf_filter$bpar_1_3m1 + dgf_filter$subclin_1_3m1 +dgf_filter$mpred_1_3m1 +dgf_filter$steroid_res_1_3m1 + dgf_filter$bkvan_1_3m1 
       +dgf_filter$bpar_3m1 + dgf_filter$subclin_3m1 +dgf_filter$mpred_3m1 +dgf_filter$steroid_res_3m1 + dgf_filter$bkvan_3m1     
       +dgf_filter$bpar_3_12m1 + dgf_filter$subclin_3_12m1 +dgf_filter$mpred_3_12m1 +dgf_filter$steroid_res_3_12m1 + dgf_filter$bkvan_3_12m1
       +dgf_filter$bpar_12m1 + dgf_filter$subclin_12m1 +dgf_filter$mpred_12m1 +dgf_filter$steroid_res_12m1 + dgf_filter$bkvan_12m1
       +dgf_filter$bpar_post12m1 + dgf_filter$subclin_post12m1 +dgf_filter$mpred_post12m1 +dgf_filter$steroid_res_post12m1 + dgf_filter$bkvan_post12m1 +dgf_filter$bpar_any1 + dgf_filter$subclin_any1 + dgf_filter$bkvan_any1 +dgf_filter$mpred_any1 +dgf_filter$steroidres_any1 + dgf_filter$IFTA_score_seb_3m_ms + dgf_filter$iIFTA_score_12m_ms| dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

table1(~dgf_filter$dgf_criteria1 + dgf_filter$donor_criteria_combined | dgf_filter$bpar_any1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
bpar_0_1dgf<- chisq.test(dgf_filter$bpar_0_1m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_1dgf<- chisq.test(dgf_filter$bpar_1m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_1_3dgf<- chisq.test(dgf_filter$bpar_1_3m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_3dgf<- chisq.test(dgf_filter$bpar_3m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_3_12dgf<- chisq.test(dgf_filter$bpar_3_12m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_12dgf<- chisq.test(dgf_filter$bpar_12m1, dgf_filter$dgf_criteria, correct = FALSE)
bpar_post12dgf<- chisq.test(dgf_filter$bpar_post12m1, dgf_filter$dgf_criteria, correct = FALSE)

Subclin_0_1dgf<- chisq.test(dgf_filter$Subclin_0_1m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_1dgf<- chisq.test(dgf_filter$subclin_1m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_1_3dgf<- chisq.test(dgf_filter$subclin_1_3m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_3dgf<- chisq.test(dgf_filter$subclin_3m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_3_12dgf<- chisq.test(dgf_filter$subclin_3_12m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_12dgf<- chisq.test(dgf_filter$subclin_12m1, dgf_filter$dgf_criteria, correct = FALSE)
subclin_post12dgf<- chisq.test(dgf_filter$subclin_post12m1, dgf_filter$dgf_criteria, correct = FALSE)

mpred_0_1dgf<- chisq.test(dgf_filter$mpred_0_1m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_1dgf<- chisq.test(dgf_filter$mpred_1m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_1_3dgf<- chisq.test(dgf_filter$mpred_1_3m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_3dgf<- chisq.test(dgf_filter$mpred_3m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_3_12dgf<- chisq.test(dgf_filter$mpred_3_12m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_12dgf<- chisq.test(dgf_filter$mpred_12m1, dgf_filter$dgf_criteria, correct = FALSE)
mpred_post12dgf<- chisq.test(dgf_filter$mpred_post12m1, dgf_filter$dgf_criteria, correct = FALSE)

steroid_res_0_1dgf<- chisq.test(dgf_filter$steroid_res_0_1m1, dgf_filter$dgf_criteria, correct = FALSE)
steroid_res_1dgf<- chisq.test(dgf_filter$steroid_res_1m1, dgf_filter$dgf_criteria, correct = FALSE)
steroid_res_1_3dgf<- chisq.test(dgf_filter$steroid_res_1_3m1, dgf_filter$dgf_criteria, correct = FALSE)
steroid_res_3dgf<- chisq.test(dgf_filter$steroid_res_3m1, dgf_filter$dgf_criteria, correct = FALSE)
steroid_res_3_12dgf<- chisq.test(dgf_filter$steroid_res_3_12m1, dgf_filter$dgf_criteria, correct = FALSE)
#steroid_res_12dgf<- chisq.test(dgf_filter$steroid_res_12m, dgf_filter$dgf_hd_criteria, correct = FALSE)
steroid_res_post12dgf<- chisq.test(dgf_filter$steroid_res_post12m1, dgf_filter$dgf_criteria, correct = FALSE)

#bkvan_0_1dgf<- chisq.test(dgf_filter$bkvan_0_1m, dgf_filter$dgf_hd_criteria, correct = FALSE) #fix
#bkvan_1dgf<- chisq.test(dgf_filter$bkvan_1m, dgf_filter$dgf_hd_criteria, correct = FALSE) #fix
bkvan_1_3dgf<- chisq.test(dgf_filter$bkvan_1_3m1, dgf_filter$dgf_criteria, correct = FALSE)
bkvan_3dgf<- chisq.test(dgf_filter$bkvan_3m1, dgf_filter$dgf_criteria, correct = FALSE)
bkvan_3_12dgf<- chisq.test(dgf_filter$bkvan_3_12m1, dgf_filter$dgf_criteria, correct = FALSE)
bkvan_12dgf<- chisq.test(dgf_filter$bkvan_12m1, dgf_filter$dgf_criteria, correct = FALSE)
#bkvan_post12dgf<- chisq.test(dgf_filter$bkvan_post12m, dgf_filter$dgf_hd_criteria, correct = FALSE)

# summarise the p.values for graft bchem by timepoints
  p.graft.reject.var <-c("BPAR (0-1m)","Subclin Rejection (0-1m)", "Mpred (0-1m)", "Steroid Res (0-1m)", 
                         "BPAR (1m)","Subclin Rejection (1m)", "Mpred (1m)", "Steroid Res (1m)",
                         "BPAR (1-3m)","Subclin Rejection (1-3m)", "Mpred (1-3m)", "Steroid Res (1-3m)", "BKVAN (1-3m)", 
                         "BPAR (3m)","Subclin Rejection (3m)", "Mpred (3m)", "Steroid Res (3m)", "BKVAN (3m)", 
                         "BPAR (3-12m)","Subclin Rejection (3-12m)", "Mpred (3-12m)", "Steroid Res (3-12m)", "BKVAN (3-12m)",
                         "BPAR (12m)","Subclin Rejection (12m)", "Mpred (12m)", "BKVAN (12m)", 
                         "BPAR (post12m)","Subclin Rejection (post12m)", "Mpred (post12m)", "Steroid Res (post12m)")
                       
  p.graft.reject.dgf<-c(bpar_0_1dgf$p.value, Subclin_0_1dgf$p.value, mpred_0_1dgf$p.value, steroid_res_0_1dgf$p.value,
                        bpar_1dgf$p.value, subclin_1dgf$p.value, mpred_1dgf$p.value, steroid_res_1dgf$p.value, bpar_1_3dgf$p.value,
                        subclin_1_3dgf$p.value,mpred_1_3dgf$p.value, steroid_res_1_3dgf$p.value, bkvan_1_3dgf$p.value,
                        bpar_3dgf$p.value, subclin_3dgf$p.value,mpred_3dgf$p.value, steroid_res_3dgf$p.value, bkvan_3dgf$p.value,
                        bpar_3_12dgf$p.value, subclin_3_12dgf$p.value,mpred_3_12dgf$p.value, steroid_res_3_12dgf$p.value,
                        bkvan_3_12dgf$p.value, bpar_12dgf$p.value, subclin_12dgf$p.value,mpred_12dgf$p.value, bkvan_12dgf$p.value,
                        bpar_post12dgf$p.value, subclin_post12dgf$p.value,mpred_post12dgf$p.value,
                        steroid_res_post12dgf$p.value,steroid_res_post12dgf$p.value)
    
  p.graft.reject.dgf<-round(p.graft.reject.dgf, digits = 3)
  p.graft.reject1 <- cbind(p.graft.reject.var, p.graft.reject.dgf)
    htmlTable((p.graft.reject1))
    
# Rejection at any time
    table1(~dgf_filter$bpar_any1 + dgf_filter$subclin_any1 + dgf_filter$bkvan_any1 +dgf_filter$mpred_any1 +dgf_filter$steroidres_any1
           |dgf_filter$dgf_criteria1, data=dgf_filter,overall="Total", 
           render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))
    
    bpar_anydgf<- chisq.test(dgf_filter$bpar_any, dgf_filter$dgf_criteria, correct = FALSE)
    subclin_anydgf<- chisq.test(dgf_filter$subclin_any, dgf_filter$dgf_criteria, correct = FALSE)
    bkvan_anydgf<- chisq.test(dgf_filter$bkvan_any, dgf_filter$dgf_criteria, correct = FALSE)
    mpred_anydgf<- chisq.test(dgf_filter$mpred_any, dgf_filter$dgf_criteria, correct = FALSE)
    steroidres_anydgf<- chisq.test(dgf_filter$steroidres_any, dgf_filter$dgf_criteria, correct = FALSE)
    
    bpar_anydgf$p.value
    subclin_anydgf$p.value
    bkvan_anydgf$p.value
    mpred_anydgf$p.value
    steroidres_anydgf$p.value
    
    p.graft.reject_any.var <-c("BPAR", "Subclin", "BKVAN", "Mpred", "Additional IS")
    p.graft.reject_any.dgf<-c(bpar_anydgf$p.value, subclin_anydgf$p.value, bkvan_anydgf$p.value, mpred_anydgf$p.value,
                              steroidres_anydgf$p.value)
    p.graft.reject_any.dgf<-round(p.graft.reject_any.dgf, digits = 3)
    p.graft.reject2 <- cbind(p.graft.reject_any.var, p.graft.reject_any.dgf)
    htmlTable((p.graft.reject2))
```


```{r, echo=FALSE, results = FALSE}
#############################
# Outcome data by DGF (CADI)
############################

# descriptive table
table1(~dgf_filter$cadi_1m + dgf_filter$cadi_3m + dgf_filter$cadi_12m| dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
cadi1m_dgf<- kruskal.test(dgf_filter$cadi_1m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cadi3m_dgf<- kruskal.test(dgf_filter$cadi_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)
cadi12m_dgf<- kruskal.test(dgf_filter$cadi_12m ~ dgf_filter$donor_criteria_base1, data = dgf_filter)

# summarise the p.values for graft bchem
  p.graft.cadi.var <-c("CADI (1m)","CADI(3m))", "CADI (12m)")
                       
  p.graft.cadi.dgf<-c(cadi1m_dgf$p.value, cadi3m_dgf$p.value, cadi12m_dgf$p.value)
    
  p.graft.cadi.dgf<-round(p.graft.cadi.dgf, digits = 3)
  p.graft.cadi1 <- cbind(p.graft.cadi.var, p.graft.cadi.dgf)
    htmlTable((p.graft.cadi1))
    
# CADI scores stratified by level = 2
    dgf_filter$cadi1m_2 <- ifelse(dgf_filter$cadi_1m >=2, 1,0)
    dgf_filter$cadi1m_2 <- as.factor(dgf_filter$cadi1m_2)
    dgf_filter$cadi3m_2 <- ifelse(dgf_filter$cadi_3m >=2, 1,0)
    dgf_filter$cadi3m_2 <- as.factor(dgf_filter$cadi3m_2)
    dgf_filter$cadi12m_2 <- ifelse(dgf_filter$cadi_12m >=2, 1,0)
    dgf_filter$cadi12m_2 <- as.factor(dgf_filter$cadi12m_2)
    
    table1(~dgf_filter$cadi1m_2 + dgf_filter$cadi3m_2 + dgf_filter$cadi12m_2| dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))
    
    cadi1m2_dgf<- chisq.test(dgf_filter$cadi1m_2, dgf_filter$dgf_criteria, correct = FALSE)
    cadi3m2_dgf<- chisq.test(dgf_filter$cadi3m_2, dgf_filter$dgf_criteria, correct = FALSE)
    cadi12m2_dgf<- chisq.test(dgf_filter$cadi12m_2, dgf_filter$dgf_criteria, correct = FALSE)

# summarise the p.values for graft bchem
  p.graft.cadi.var <-c("CADI (1m)","CADI(3m))", "CADI (12m)")
                       
  p.graft.cadi.dgf<-c(cadi1m2_dgf$p.value, cadi3m2_dgf$p.value, cadi12m2_dgf$p.value)
    
  p.graft.cadi.dgf<-round(p.graft.cadi.dgf, digits = 3)
  p.graft.cadi2 <- cbind(p.graft.cadi.var, p.graft.cadi.dgf)
    htmlTable((p.graft.cadi2))
    
    # fishers
    c1m_2<-xtabs( ~dgf_criteria + cadi1m_2, dgf_filter)
    fisher.test(c1m_2)
    
    c3m_2<-xtabs( ~dgf_criteria + cadi3m_2, dgf_filter)
    fisher.test(c3m_2)
    
    c12m_2<-xtabs( ~dgf_criteria + cadi12m_2, dgf_filter)
    fisher.test(c12m_2)
    
```
 
    
```{r}
###########
# Outcomes - complications
##########

dgf_filter$nodat<-as.factor(dgf_filter$nodat)

table1(~dgf_filter$cmv_viremia +dgf_filter$cmv_disease + dgf_filter$ebv_infection + dgf_filter$bk_viremia +dgf_filter$bkvan +dgf_filter$fungal_invasive_orblood
       +dgf_filter$PTLD +dgf_filter$skin_ca + dgf_filter$other_ca + dgf_filter$cvd_0_5y + dgf_filter$nodat +dgf_filter$gasteroenteritis+ dgf_filter$chest_infection + dgf_filter$recurrent_uti + dgf_filter$resistant_uti | 
         dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total",
       render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

    cmv1<-chisq.test(dgf_filter$cmv_viremia, dgf_filter$dgf_criteria, correct = FALSE)
    cmv2<-chisq.test(dgf_filter$cmv_disease, dgf_filter$dgf_criteria, correct = FALSE)
    ebv<-chisq.test(dgf_filter$ebv_infection, dgf_filter$dgf_criteria, correct = FALSE)
    bkvir<-chisq.test(dgf_filter$bk_viremia, dgf_filter$dgf_criteria, correct = FALSE)
    bkva<-chisq.test(dgf_filter$bkvan, dgf_filter$dgf_criteria, correct = FALSE)
    fungal<-chisq.test(dgf_filter$fungal_invasive_orblood, dgf_filter$dgf_criteria, correct = FALSE)
    ptld<-chisq.test(dgf_filter$PTLD, dgf_filter$dgf_criteria, correct = FALSE)
    skinca<-chisq.test(dgf_filter$skin_ca, dgf_filter$dgf_criteria, correct = FALSE)
    otherca<-chisq.test(dgf_filter$other_ca, dgf_filter$dgf_criteria, correct = FALSE)
    cvd<-chisq.test(dgf_filter$cvd_0_5y, dgf_filter$dgf_criteria, correct = FALSE)
    nodat<-chisq.test(dgf_filter$nodat, dgf_filter$dgf_criteria, correct = FALSE)
    gastro<-chisq.test(dgf_filter$gasteroenteritis, dgf_filter$dgf_criteria, correct = FALSE)
    chest<-chisq.test(dgf_filter$chest_infection, dgf_filter$dgf_criteria, correct = FALSE)
    recuti<-chisq.test(dgf_filter$recurrent_uti, dgf_filter$dgf_criteria, correct = FALSE)
    resuti<-chisq.test(dgf_filter$resistant_uti, dgf_filter$dgf_criteria, correct = FALSE)

    complications<-c(cmv1$p.value, cmv2$p.value, ebv$p.value, bkvir$p.value, bkva$p.value, fungal$p.value, ptld$p.value, skinca$p.value, otherca$p.value, cvd$p.value,nodat$p.value, gastro$p.value, chest$p.value, recuti$p.value, resuti$p.value)
    complications<-round(complications, 3)
    View(complications)
```


```{r}
##########################
# NEUTROPHILS 
##########################

table1(~dgf_filter$neut_0m+dgf_filter$neut_1m+dgf_filter$neut_3m|dgf_filter$dgf_criteria1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

table1(~dgf_filter$neut_0m+dgf_filter$neut_1m+dgf_filter$neut_3m|dgf_filter$donor_criteria_ECD1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD%)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# diff between DGF groups by HD criteria
neut0m<- kruskal.test(dgf_filter$neut_0m ~ dgf_filter$dgf_criteria, data = dgf_filter)
neut1m<- kruskal.test(dgf_filter$neut_1m ~ dgf_filter$dgf_criteria, data = dgf_filter)
neut3m<- kruskal.test(dgf_filter$neut_3m ~ dgf_filter$dgf_criteria, data = dgf_filter)

neut0m1<- kruskal.test(dgf_filter$neut_0m ~ dgf_filter$donor_criteria_combined, data = dgf_filter)
neut1m1<- kruskal.test(dgf_filter$neut_1m ~ dgf_filter$donor_criteria_combined, data = dgf_filter)
neut3m1<- kruskal.test(dgf_filter$neut_3m ~ dgf_filter$donor_criteria_combined, data = dgf_filter)

# summarise the p.values for graft bchem
  p.graft.neut.var <-c("Neutrophils (0m)","Neutrophils (1m))", "Neutrophils (3m)")
                       
  p.graft.neut.dgf<-c(neut0m$p.value, neut1m$p.value, neut3m$p.value)
  p.graft.neut.donor<-c(neut0m1$p.value, neut1m1$p.value, neut3m1$p.value)
  p.graft.neut.dgf<-round(p.graft.neut.dgf, digits = 3)
  p.graft.neut.donor<-round(p.graft.neut.donor, digits = 3)
  p.graft.neut1 <- cbind(p.graft.neut.var, p.graft.neut.dgf, p.graft.neut.donor)
    htmlTable((p.graft.neut1))

    # plots
  n0<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=neut_0m, fill=dgf_criteria1, colour=dgf_criteria1))+
      geom_jitter() +
      theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("Neutrophils/hpf")+  ylim(0,15)+
    ggtitle("0 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)
    
  n1<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=neut_1m, fill=dgf_criteria1,colour=dgf_criteria1))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("1 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.98)
  
  n3<-ggplot(dgf_filter, aes(x=dgf_criteria1, y=neut_3m, fill=dgf_criteria1,colour=dgf_criteria1))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("3 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)
             
    ggarrange(n0,n1,n3, ncol=3, nrol=1, fontsize = 3)
    
    # by donor criteria
      n0a<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=neut_0m, fill=donor_criteria_base1, colour=donor_criteria_base1))+
      geom_jitter() +
      theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("Neutrophils/hpf")+  ylim(0,15)+
    ggtitle("0 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)
    
  n1a<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=neut_1m, fill=donor_criteria_base1,colour=donor_criteria_base1))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("1 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.98)
  
  n3a<-ggplot(dgf_filter, aes(x=donor_criteria_base1, y=neut_3m, fill=donor_criteria_base1, colour=donor_criteria_base1))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("3 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)
             
    neutplot1<-dgf_filter[,c("neut_0m","donor_criteria_base1", "donor_criteria_ECD1")]
    neutplot1<-melt(neutplot1,id="neut_0m")
    colnames(neutplot1)<-c("neut_0m","donor_criteria", "donor_criteria_all")
    neutplot1$rm<-ifelse(neutplot1$donor_criteria=="donor_criteria_ECD1" & neutplot1$donor_criteria_all=="ECD",1,0)
    neutplot1$neut_0m<-ifelse(neutplot1$rm ==1 | neutplot1$donor_criteria=="donor_criteria_base1", neutplot1$neut_0m, NA)
    neutplot1<-neutplot1[!is.na(neutplot1$donor_criteria_all),]
    neutplot1<-neutplot1[!is.na(neutplot1$neut_0m),]
  
    neutplot2<-dgf_filter[,c("neut_1m","donor_criteria_base1", "donor_criteria_ECD1")]
    neutplot2<-melt(neutplot2,id="neut_1m")
    colnames(neutplot2)<-c("neut_1m","donor_criteria", "donor_criteria_all")
    neutplot2$rm<-ifelse(neutplot2$donor_criteria=="donor_criteria_ECD1" & neutplot2$donor_criteria_all=="ECD",1,0)
    neutplot2$neut_1m<-ifelse(neutplot2$rm ==1 | neutplot2$donor_criteria=="donor_criteria_base1", neutplot2$neut_1m, NA)
    neutplot2<-neutplot2[!is.na(neutplot2$donor_criteria_all),]
    neutplot2<-neutplot2[!is.na(neutplot2$neut_1m),]
    
    neutplot3<-dgf_filter[,c("neut_3m","donor_criteria_base1", "donor_criteria_ECD1")]
    neutplot3<-melt(neutplot3,id="neut_3m")
    colnames(neutplot3)<-c("neut_3m","donor_criteria", "donor_criteria_all")
    neutplot3$rm<-ifelse(neutplot3$donor_criteria=="donor_criteria_ECD1" & neutplot3$donor_criteria_all=="ECD",1,0)
    neutplot3$neut_3m<-ifelse(neutplot3$rm ==1 | neutplot3$donor_criteria=="donor_criteria_base1", neutplot3$neut_3m, NA)
    neutplot3<-neutplot3[!is.na(neutplot3$donor_criteria_all),]
    neutplot3<-neutplot3[!is.na(neutplot3$neut_3m),]
    
  n0b<-ggplot(neutplot1, aes(x=donor_criteria_all, y=neut_0m, colour=donor_criteria_all, na.rm=TRUE))+
      geom_jitter() +
      theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+
      ylab("Neutrophils/hpf")+  ylim(0,15)+
    ggtitle("0 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)
    
  n1b<-ggplot(neutplot2, aes(x=donor_criteria_all, y=neut_1m, colour=donor_criteria_all, na.rm=TRUE))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("1 month")+
      stat_summary(fun = median,
                 geom = "crossbar", width = 0.6, size=0.98)
  
  n3b<-ggplot(neutplot3, aes(x=donor_criteria_all, y=neut_3m, colour=donor_criteria_all, na.rm=TRUE))+
      geom_jitter() +
    theme_bw(base_size=10)+
      theme(legend.position="none")+
      xlab("Groups")+ ylim(0,15)+
      ylab("Neutrophils/hpf")+ ggtitle("3 month")+
      stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.6, size=0.8)

        ggarrange(n0,n1,n3,n0b,n1b,n3b, ncol=3, nrow=2, fontsize = 4)

    #neutrophils split by rejection within 3 months
    dgf_filter$bparany03m<-ifelse(dgf_filter$bpar_0_1m>0 | dgf_filter$bpar_1m>0 | 
                                                 dgf_filter$bpar_1_3m>0 | dgf_filter$bpar_3m>0 ,1,0)
    dgf_filter$bparany03m<-replace_na(dgf_filter$bparany03m,0)
    dgf_filter$bparany03m<-factor(dgf_filter$bparany03m, levels = c(0,1), labels=c("No rejection", "Rejection (0-3m)"))
    
        nr1<-ggplot(dgf_filter, aes(x=bparany03m, y=neut_0m, colour=bparany03m, na.rm=TRUE))+
        geom_jitter() +
        theme_bw(base_size=10)+
        theme(legend.position="none")+
        xlab("Groups")+ ylim(0,15)+
        ylab("Neutrophils/hpf")+ ggtitle("0 month")+
        stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.3, size=0.8)
        
        nr2<-ggplot(dgf_filter, aes(x=bparany03m, y=neut_1m, colour=bparany03m, na.rm=TRUE))+
        geom_jitter() +
        theme_bw(base_size=10)+
        theme(legend.position="none")+
        xlab("Groups")+ ylim(0,15)+
        ylab("Neutrophils/hpf")+ ggtitle("1 month")+
        stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.3, size=0.8)

        nr3<-ggplot(dgf_filter, aes(x=bparany03m, y=neut_3m, colour=bparany03m, na.rm=TRUE))+
        geom_jitter() +
        theme_bw(base_size=10)+
        theme(legend.position="none")+
        xlab("Groups")+ ylim(0,15)+
        ylab("Neutrophils/hpf")+ ggtitle("3 month")+
        stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.3, size=0.8)
        
    ggarrange(n0,n1,n3,n0b,n1b,n3b,nr1, nr2, nr3, ncol=3, nrow=3, fontsize = 4)
        
```


```{r}
########################
# Multivariate Logisic Regression - Rejection (BPAR)
#######################
table1(~dgf_filter$dgf_criteria1+dgf_filter$graft_number1 +dgf_filter$HLA_mm1 +dgf_filter$dsa1_pre1 +dgf_filter$dsa2_pre1 +dgf_filter$ondialysis_pretx1 +dgf_filter$renal_dx1 +dgf_filter$donor_age +dgf_filter$donor_criteria_combined +dgf_filter$induct_extra +dgf_filter$donor_terminal_cr +dgf_filter$donor_terminal_urea +dgf_filter$cit_min +dgf_filter$anastamosis_time_mins +dgf_filter$transfuse_1yr1| dgf_filter$bpar_any1, data=dgf_filter, overall="Total", render.continuous=c(.="Mean (SD)", .="Median [Min, Max]","Median, IQR"="Median (IQR)"))

# === determinants of rejection (all)
dgf_filter$induct_extra_new<-ifelse(is.na(dgf_filter$induct_extra)==TRUE, 1,0)
set3<-glm(dgf_filter$bpar_any ~ dgf_filter$dgf_criteria1+dgf_filter$graft_number1 +dgf_filter$HLA_mm1 +dgf_filter$dsa1_pre1 +dgf_filter$dsa2_pre1 +dgf_filter$ondialysis_pretx1 +dgf_filter$renal_dx1 +dgf_filter$donor_age +dgf_filter$donor_criteria_combined +dgf_filter$donor_terminal_cr +dgf_filter$induct_extra_new +dgf_filter$donor_terminal_urea +dgf_filter$cit_min +dgf_filter$anastamosis_time_mins +dgf_filter$transfuse_1yr, family="binomial")
summary(set3)

# === determinants of rejection (select)
set4<-glm(dgf_filter$bpar_any ~ dgf_filter$dgf_criteria1+dgf_filter$induct_extra_new +dgf_filter$donor_criteria_combined, family="binomial")
summary(set4)
nagelkerke(set4)

#graft loss
set5<-glm(dgf_filter$graft_loss_kidneyonly1 ~ dgf_filter$bpar_any1+ dgf_filter$subclin_any1 + dgf_filter$bkvan_any1 +dgf_filter$dgf_criteria1+dgf_filter$graft_number1 +dgf_filter$HLA_mm1 +dgf_filter$dsa1_pre1 +dgf_filter$dsa2_pre1 +dgf_filter$ondialysis_pretx1 +dgf_filter$renal_dx1 +dgf_filter$donor_criteria_combined +dgf_filter$donor_terminal_cr +dgf_filter$donor_terminal_urea +dgf_filter$transfuse_1yr, family="binomial")
summary(set5)

# === determinants of graft loss (select)
set6<-glm(dgf_filter$graft_loss_kidneyonly1 ~ dgf_filter$bpar_any1+ dgf_filter$subclin_any1 + dgf_filter$bkvan_any1 + dgf_filter$dgf_criteria1, family="binomial")
summary(set6)
nagelkerke(set6)

set7<-glm(dgf_filter$graft_loss_kidneyonly1 ~ dgf_filter$bpar_0_1m+ dgf_filter$bpar_1_3m + dgf_filter$bpar_3m + dgf_filter$dgf_criteria1, family="binomial")
summary(set7)
nagelkerke(set7)
```


```{r}
######################
# Biopsy scores - protocol only
#####################
dgf_filter.orig = dgf_filter

dgf_filter = dgf_filter.orig
dgf_filter = dgf_filter[dgf_filter$bparupto03m == 1,]

# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0 = tapply(dgf_filter$ci_0m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.0 = tapply(dgf_filter$ci_0m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ci_0m)-sum(is.na(dgf_filter$ci_0m)))
i.sd.0 = tapply(dgf_filter$ci_0m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.0 = tapply(dgf_filter$ci_0m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.1<-tapply(dgf_filter$ci_1m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.1<-tapply(dgf_filter$ci_1m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ci_1m)-sum(is.na(dgf_filter$ci_1m)))
i.sd.1 = tapply(dgf_filter$ci_1m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.1 = tapply(dgf_filter$ci_1m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.3<-tapply(dgf_filter$ci_3m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.3<-tapply(dgf_filter$ci_3m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ci_3m)-sum(is.na(dgf_filter$ci_3m)))
i.sd.3 = tapply(dgf_filter$ci_3m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.3 = tapply(dgf_filter$ci_3m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.12<-tapply(dgf_filter$ci_12m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.12<-tapply(dgf_filter$ci_12m, dgf_filter$dgf_criteria, sd,
                 na.rm=TRUE)/sqrt(length(dgf_filter$ci_12m)-sum(is.na(dgf_filter$ci_12m)))
i.sd.12 = tapply(dgf_filter$ci_12m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.12 = tapply(dgf_filter$ci_12m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i.sd <- rbind(i.sd.0, i.sd.1, i.sd.3, i.sd.12)
protocol.i.sd<-melt(protocol.i.sd, id.var=dgf)
protocol.i.sd<-cbind(p.time, protocol.i.sd)
colnames(protocol.i.sd)<-c("time", "stat", "strata", "sd")

protocol.i.n <- rbind(i.n.0, i.n.1, i.n.3, i.n.12)
protocol.i.n<-melt(protocol.i.n, id.var=dgf)
protocol.i.n<-cbind(p.time, protocol.i.n)
colnames(protocol.i.n)<-c("time", "stat", "strata", "n")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem
protocol.i$sd<-protocol.i.sd$sd
protocol.i$n<-protocol.i.n$n

ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sd), ymax=(mean+sd), group=strata),  width=.2) +
  theme_classic() +
  ggtitle("Banff-ci (BPAR 0-3m)")+
  scale_x_continuous(breaks = c(0,1,3,12))+
  xlab("Time (months)")+
  ylab("Banff-ci score")

d0m = fisher.test(table(dgf_filter$ci_0m, dgf_filter$dgf_criteria))
d1m = fisher.test(table(dgf_filter$ci_1m, dgf_filter$dgf_criteria))
d3m = fisher.test(table(dgf_filter$ci_3m, dgf_filter$dgf_criteria))
d12m = fisher.test(table(dgf_filter$ci_12m, dgf_filter$dgf_criteria))

c(d0m$p.value, d1m$p.value, d3m$p.value, d12m$p.value)
```

```{r}
######################
# Biopsy scores - protocol only
#####################
dgf_filter = dgf_filter.orig
dgf_filter = dgf_filter[dgf_filter$bparupto03m == 0,]

# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_0m)-sum(is.na(dgf_filter$ct_0m)))
i.sd.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.1<-tapply(dgf_filter$ct_1m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.1<-tapply(dgf_filter$ct_1m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_1m)-sum(is.na(dgf_filter$ct_1m)))
i.sd.1 = tapply(dgf_filter$ct_1m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.1 = tapply(dgf_filter$ct_1m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.3<-tapply(dgf_filter$ct_3m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.3<-tapply(dgf_filter$ct_3m, dgf_filter$dgf_criteria, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_3m)-sum(is.na(dgf_filter$ct_3m)))
i.sd.3 = tapply(dgf_filter$ct_3m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.3 = tapply(dgf_filter$ct_3m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

i.mean.12<-tapply(dgf_filter$ct_12m, dgf_filter$dgf_criteria, mean, na.rm=TRUE)
i.sem.12<-tapply(dgf_filter$ct_12m, dgf_filter$dgf_criteria, sd,
                 na.rm=TRUE)/sqrt(length(dgf_filter$ct_12m)-sum(is.na(dgf_filter$ct_12m)))
i.sd.12 = tapply(dgf_filter$ct_12m, dgf_filter$dgf_criteria, sd, na.rm = TRUE)
i.n.12 = tapply(dgf_filter$ct_12m, dgf_filter$dgf_criteria, count, na.rm=TRUE)

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i.sd <- rbind(i.sd.0, i.sd.1, i.sd.3, i.sd.12)
protocol.i.sd<-melt(protocol.i.sd, id.var=dgf)
protocol.i.sd<-cbind(p.time, protocol.i.sd)
colnames(protocol.i.sd)<-c("time", "stat", "strata", "sd")

protocol.i.n <- rbind(i.n.0, i.n.1, i.n.3, i.n.12)
protocol.i.n<-melt(protocol.i.n, id.var=dgf)
protocol.i.n<-cbind(p.time, protocol.i.n)
colnames(protocol.i.n)<-c("time", "stat", "strata", "n")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem
protocol.i$sd<-protocol.i.sd$sd
protocol.i$n<-protocol.i.n$n

ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sd), ymax=(mean+sd), group=strata),  width=.2) +
  theme_classic() +
  ggtitle("Banff-ct (no BPAR 0-3m)")+
  scale_x_continuous(breaks = c(0,1,3,12))+
  xlab("Time (months)")+
  ylab("Banff-ct score")

d0m = fisher.test(table(dgf_filter$ct_0m, dgf_filter$dgf_criteria))
d1m = fisher.test(table(dgf_filter$ct_1m, dgf_filter$dgf_criteria))
d3m = fisher.test(table(dgf_filter$ct_3m, dgf_filter$dgf_criteria))
d12m = fisher.test(table(dgf_filter$ct_12m, dgf_filter$dgf_criteria))

c(d0m$p.value, d1m$p.value, d3m$p.value, d12m$p.value)
```


```{r}
######################
# Biopsy scores - protocol only
#####################
dgf_filter1 = dgf_filter[dgf_filter$bparupto03m == 0, ]
                         
# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0 = tapply(dgf_filter1$ct_0m, dgf_filter1$dgf_severity, mean, na.rm=TRUE)
i.sem.0 = tapply(dgf_filter1$ct_0m, dgf_filter1$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter1$ct_0m)-sum(is.na(dgf_filter1$ct_0m)))
i.sd.0 = tapply(dgf_filter1$ct_0m, dgf_filter1$dgf_severity, sd, na.rm = TRUE)
i.n.0 = tapply(dgf_filter1$ct_0m, dgf_filter1$dgf_severity, count, na.rm=TRUE)

i.mean.1<-tapply(dgf_filter1$ct_1m, dgf_filter1$dgf_severity, mean, na.rm=TRUE)
i.sem.1<-tapply(dgf_filter1$ct_1m, dgf_filter1$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter1$ct_1m)-sum(is.na(dgf_filter1$ct_1m)))
i.sd.1 = tapply(dgf_filter1$ct_1m, dgf_filter1$dgf_severity, sd, na.rm = TRUE)
i.n.1 = tapply(dgf_filter1$ct_1m, dgf_filter1$dgf_severity, count, na.rm=TRUE)

i.mean.3<-tapply(dgf_filter1$ct_3m, dgf_filter1$dgf_severity, mean, na.rm=TRUE)
i.sem.3<-tapply(dgf_filter1$ct_3m, dgf_filter1$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter1$ct_3m)-sum(is.na(dgf_filter1$ct_3m)))
i.sd.3 = tapply(dgf_filter1$ct_3m, dgf_filter1$dgf_severity, sd, na.rm = TRUE)
i.n.3 = tapply(dgf_filter1$ct_3m, dgf_filter1$dgf_severity, count, na.rm=TRUE)

i.mean.12<-tapply(dgf_filter1$ct_12m, dgf_filter1$dgf_severity, mean, na.rm=TRUE)
i.sem.12<-tapply(dgf_filter1$ct_12m, dgf_filter1$dgf_severity, sd,
                 na.rm=TRUE)/sqrt(length(dgf_filter1$ct_12m)-sum(is.na(dgf_filter$ct_12m)))
i.sd.12 = tapply(dgf_filter1$ct_12m, dgf_filter1$dgf_severity, sd, na.rm = TRUE)
i.n.12 = tapply(dgf_filter1$ct_12m, dgf_filter1$dgf_severity, count, na.rm=TRUE)

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i.sd <- rbind(i.sd.0, i.sd.1, i.sd.3, i.sd.12)
protocol.i.sd<-melt(protocol.i.sd, id.var=dgf)
protocol.i.sd<-cbind(p.time, protocol.i.sd)
colnames(protocol.i.sd)<-c("time", "stat", "strata", "sd")

protocol.i.n <- rbind(i.n.0, i.n.1, i.n.3, i.n.12)
protocol.i.n<-melt(protocol.i.n, id.var=dgf)
protocol.i.n<-cbind(p.time, protocol.i.n)
colnames(protocol.i.n)<-c("time", "stat", "strata", "n")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem
protocol.i$sd<-protocol.i.sd$sd
protocol.i$n<-protocol.i.n$n

ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  theme_classic() +
  ggtitle("Banff-ct (no BPAR 0-3m)")+
  scale_x_continuous(breaks = c(0,1,3,12))+
  xlab("Time (months)")+
  ylab("Banff-ct score")

# d0m = fisher.test(table(dgf_filter$ci_0m, dgf_filter$dgf_severity))
# d1m = fisher.test(table(dgf_filter$ci_1m, dgf_filter$dgf_severity))
# d3m = fisher.test(table(dgf_filter$ci_3m, dgf_filter$dgf_severity))
# d12m = fisher.test(table(dgf_filter$ci_12m, dgf_filter$dgf_severity))
# 
# c(d0m$p.value, d1m$p.value, d3m$p.value, d12m$p.value)
```

```{r}
######################
# Biopsy scores - protocol only
#####################

# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_severity, mean, na.rm=TRUE)
i.sem.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_0m)-sum(is.na(dgf_filter$ct_0m)))
i.sd.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_severity, sd, na.rm = TRUE)
i.n.0 = tapply(dgf_filter$ct_0m, dgf_filter$dgf_severity, count, na.rm=TRUE)

i.mean.1<-tapply(dgf_filter$ct_1m, dgf_filter$dgf_severity, mean, na.rm=TRUE)
i.sem.1<-tapply(dgf_filter$ct_1m, dgf_filter$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_1m)-sum(is.na(dgf_filter$ct_1m)))
i.sd.1 = tapply(dgf_filter$ct_1m, dgf_filter$dgf_severity, sd, na.rm = TRUE)
i.n.1 = tapply(dgf_filter$ct_1m, dgf_filter$dgf_severity, count, na.rm=TRUE)

i.mean.3<-tapply(dgf_filter$ct_3m, dgf_filter$dgf_severity, mean, na.rm=TRUE)
i.sem.3<-tapply(dgf_filter$ct_3m, dgf_filter$dgf_severity, sd,
                na.rm=TRUE)/sqrt(length(dgf_filter$ct_3m)-sum(is.na(dgf_filter$ct_3m)))
i.sd.3 = tapply(dgf_filter$ct_3m, dgf_filter$dgf_severity, sd, na.rm = TRUE)
i.n.3 = tapply(dgf_filter$ct_3m, dgf_filter$dgf_severity, count, na.rm=TRUE)

i.mean.12<-tapply(dgf_filter$ct_12m, dgf_filter$dgf_severity, mean, na.rm=TRUE)
i.sem.12<-tapply(dgf_filter$ct_12m, dgf_filter$dgf_severity, sd,
                 na.rm=TRUE)/sqrt(length(dgf_filter$ct_12m)-sum(is.na(dgf_filter$ct_12m)))
i.sd.12 = tapply(dgf_filter$ct_12m, dgf_filter$dgf_severity, sd, na.rm = TRUE)
i.n.12 = tapply(dgf_filter$ct_12m, dgf_filter$dgf_severity, count, na.rm=TRUE)

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i.sd <- rbind(i.sd.0, i.sd.1, i.sd.3, i.sd.12)
protocol.i.sd<-melt(protocol.i.sd, id.var=dgf)
protocol.i.sd<-cbind(p.time, protocol.i.sd)
colnames(protocol.i.sd)<-c("time", "stat", "strata", "sd")

protocol.i.n <- rbind(i.n.0, i.n.1, i.n.3, i.n.12)
protocol.i.n<-melt(protocol.i.n, id.var=dgf)
protocol.i.n<-cbind(p.time, protocol.i.n)
colnames(protocol.i.n)<-c("time", "stat", "strata", "n")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem
protocol.i$sd<-protocol.i.sd$sd
protocol.i$n<-protocol.i.n$n

ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  theme_classic() +
  ggtitle("Banff-ct (protocol biopsies)")+
  scale_x_continuous(breaks = c(0,1,3,12))+
  xlab("Time (months)")+
  ylab("Banff-ct score")

d0m = fisher.test(table(dgf_filter$ct_0m, dgf_filter$dgf_criteria))
d1m = fisher.test(table(dgf_filter$ct_1m, dgf_filter$dgf_criteria))
d3m = fisher.test(table(dgf_filter$ct_3m, dgf_filter$dgf_criteria))
d12m = fisher.test(table(dgf_filter$ct_12m, dgf_filter$dgf_criteria))

c(d0m$p.value, d1m$p.value, d3m$p.value, d12m$p.value)
```


```{r}
####################
# PROTOCOL KINETICS - DGF criteria
###################

# overall cohort versus stratify by variable - eg rejection
r.set<-dgf_filter[which(dgf_filter$donor_criteria_ECD==1),]

# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0<-tapply(r.set$i_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.0<-tapply(r.set$i_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_0m)-sum(is.na(r.set$i_0m)))
i.mean.1<-tapply(r.set$i_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.1<-tapply(r.set$i_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_1m)-sum(is.na(r.set$i_1m)))
i.mean.3<-tapply(r.set$i_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.3<-tapply(r.set$i_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_3m)-sum(is.na(r.set$i_3m)))
i.mean.12<-tapply(r.set$i_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.12<-tapply(r.set$i_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_12m)-sum(is.na(r.set$i_12m)))

# itest1<- t.test(r.set$i_1m~r.set$dgf_criteria)
# itest3<- t.test(r.set$i_3m~r.set$dgf_criteria)
# itest12<- t.test(r.set$i_12m~r.set$dgf_criteria)
# #itest0<- t.test(r.set$i_0m~r.set$dgf_criteria)
# i.ttest<-c(itest1$p.value, itest3$p.value, itest12$p.value) 

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem

# t-score
t.mean.0<-tapply(r.set$t_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.0<-tapply(r.set$t_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_0m)-sum(is.na(r.set$t_0m)))
t.mean.1<-tapply(r.set$t_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.1<-tapply(r.set$t_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_1m)-sum(is.na(r.set$t_1m)))
t.mean.3<-tapply(r.set$t_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.3<-tapply(r.set$t_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_3m)-sum(is.na(r.set$t_3m)))
t.mean.12<-tapply(r.set$t_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.12<-tapply(r.set$t_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_12m)-sum(is.na(r.set$t_12m)))

# ttest1<- t.test(r.set$t_1m~r.set$dgf_criteria)
# ttest3<- t.test(r.set$t_3m~r.set$dgf_criteria)
# ttest12<- t.test(r.set$t_12m~r.set$dgf_criteria)
# #ttest0<- t.test(r.set$t_0m~r.set$dgf_criteria)
# t.ttest<-c(ttest1$p.value, ttest3$p.value, ttest12$p.value) #

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.t.mean<- rbind(t.mean.0, t.mean.1, t.mean.3, t.mean.12)
protocol.t.mean<-melt(protocol.t.mean,id.var= dgf)
protocol.t.mean<-cbind(p.time, protocol.t.mean)
colnames(protocol.t.mean)<-c("time", "stat", "strata", "mean")

protocol.t.sem <- rbind(t.sem.0, t.sem.1, t.sem.3, t.sem.12)
protocol.t.sem<-melt(protocol.t.sem, id.var=dgf)
protocol.t.sem<-cbind(p.time, protocol.t.sem)
colnames(protocol.t.sem)<-c("time", "stat", "strata", "sem")

protocol.t<-protocol.t.mean
protocol.t<-as.data.frame(protocol.t)
protocol.t$sem<-protocol.t.sem$sem

# ci score
ci.mean.0<-tapply(r.set$ci_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.0<-tapply(r.set$ci_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_0m)-sum(is.na(r.set$ci_0m)))
ci.mean.1<-tapply(r.set$ci_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.1<-tapply(r.set$ci_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_1m)-sum(is.na(r.set$ci_1m)))
ci.mean.3<-tapply(r.set$ci_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.3<-tapply(r.set$ci_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_3m)-sum(is.na(r.set$ci_3m)))
ci.mean.12<-tapply(r.set$ci_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.12<-tapply(r.set$ci_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_12m)-sum(is.na(r.set$ci_12m)))
#
# citest1<- t.test(r.set$ci_1m~r.set$dgf_criteria)
# citest3<- t.test(r.set$ci_3m~r.set$dgf_criteria)
# citest12<- t.test(r.set$ci_12m~r.set$dgf_criteria)
# #citest0<- t.test(r.set$ci_0m~r.set$dgf_criteria)
# ci.ttest<-c(citest1$p.value, citest3$p.value, citest12$p.value) #

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.ci.mean<- rbind(ci.mean.0, ci.mean.1, ci.mean.3, ci.mean.12)
protocol.ci.mean<-melt(protocol.ci.mean, id.var=dgf)
protocol.ci.mean<-cbind(p.time, protocol.ci.mean)
colnames(protocol.ci.mean)<-c("time", "stat", "strata", "mean")

protocol.ci.sem <- rbind(ci.sem.0, ci.sem.1, ci.sem.3, ci.sem.12)
protocol.ci.sem<-melt(protocol.ci.sem, id.var=dgf)
protocol.ci.sem<-cbind(p.time, protocol.ci.sem)
colnames(protocol.ci.sem)<-c("time", "stat", "strata", "sem")

protocol.ci<-protocol.ci.mean
protocol.ci<-as.data.frame(protocol.ci)
protocol.ci$sem<-protocol.ci.sem$sem

# ct score
ct.mean.0<-tapply(r.set$ct_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.0<-tapply(r.set$ct_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_0m)-sum(is.na(r.set$ct_0m)))
ct.mean.1<-tapply(r.set$ct_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.1<-tapply(r.set$ct_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_1m)-sum(is.na(r.set$ct_1m)))
ct.mean.3<-tapply(r.set$ct_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.3<-tapply(r.set$ct_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_3m)-sum(is.na(r.set$ct_3m)))
ct.mean.12<-tapply(r.set$ct_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.12<-tapply(r.set$ct_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_12m)-sum(is.na(r.set$ct_12m)))

# cttest1<- t.test(r.set$ct_1m~r.set$dgf_criteria)
# cttest3<- t.test(r.set$ct_3m~r.set$dgf_criteria)
# cttest12<- t.test(r.set$ct_12m~r.set$dgf_criteria)
# #cttest0<- t.test(r.set$ct_0m~r.set$dgf_criteria)
# ct.ttest<-c( cttest1$p.value, cttest3$p.value, cttest12$p.value) #

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1)

protocol.ct.mean<- rbind(ct.mean.0, ct.mean.1, ct.mean.3, ct.mean.12)
protocol.ct.mean<-melt(protocol.ct.mean, id.var=dgf)
protocol.ct.mean<-cbind(p.time, protocol.ct.mean)
colnames(protocol.ct.mean)<-c("time", "stat", "strata", "mean")

protocol.ct.sem <- rbind(ct.sem.0, ct.sem.1, ct.sem.3, ct.sem.12)
protocol.ct.sem<-melt(protocol.ct.sem, id.var=dgf)
protocol.ct.sem<-cbind(p.time, protocol.ct.sem)
colnames(protocol.ct.sem)<-c("time", "stat", "strata", "sem")

protocol.ct<-protocol.ct.mean
protocol.ct<-as.data.frame(protocol.ct)
protocol.ct$sem<-protocol.ct.sem$sem

# table1(~ci_0m+ct_0m +ci_1m+ct_1m + ci_3m+ct_3m+ ci_12m+ct_12m|dgf_criteria1, data = r.set)
# statp<-cbind(ci.ttest, ct.ttest)
# View(statp)

# plots
i.score<-
  ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
  ylim(0,1.8)+
  ggtitle("Banff-i")+
  xlab("Time (months)")+
  ylab("Banff-i score")

ci.score<-
  ggplot(protocol.ci, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
  ylim(0,1.8)+
  xlim(0,12)+
  ggtitle("Banff-ci (ECD)")+
  xlab("Time (months)")+
  ylab("Banff-ci score")

t.score<-
  ggplot(protocol.t, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
    ylim(0,1.8)+
  ggtitle("Banff-t")+
  xlab("Time (months)")+
  ylab("Banff-t score")

ct.score<-
  ggplot(protocol.ct, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
  ylim(0,1.8)+
  ggtitle("Banff-ct (ECD)")+
  xlab("Time (months)")+
  ylab("Banff-ct score")

ggarrange(i.score, t.score, ci.score, ct.score)

```


```{r}

ci.all<-ci.score
ct.all<-ct.score

ci.dcd<-ci.score
ct.dcd<-ct.score

ci.ecd<-ci.score
ct.ecd<-ct.score

ci.bp<-ci.score
ct.bp<-ct.score

ci.bpfree<-ci.score
ct.bpfree<-ct.score

ci.sub<-ci.score
ct.sub<-ct.score

ci.sub01<-ci.score
ct.sub01<-ct.score

ci.aki<-ci.score
ct.aki<-ct.score

ggarrange(ci.all, ci.bp, ci.dcd, ct.dcd, ct.ecd, ct.ecd,nrow=2, ncol=3)
ggarrange(ci.all, ci.bp, ci.sub, ct.all, ct.bp, ct.sub,nrow=2, ncol=3)
ggarrange(ci.all, ci.bp, ci.sub, ci.sub01, ct.all, ct.bp, ct.sub, ct.sub01, nrow=2, ncol=4)

ci.aki.all <-ci.score
ct.aki.all <-ct.score

ci.aki.bp <-ci.score
ct.aki.bp <-ct.score


ggarrange(ci.all, ci.bp, ci.aki.all, ci.aki.bp, ct.all, ct.bp, ct.aki.all, ct.aki.bp, nrow=2, ncol=4)

ggarrange(ci.dcd, ci.ecd, ci.aki.all, ci.aki.bp, ct.dcd, ct.ecd, ct.aki.all, ct.aki.bp, nrow=2, ncol=4)

ggarrange(ci.all, ci.bp, ci.dcd, ci.sub,
          ct.all, ct.bp, ct.dcd, ct.sub,
          ci.aki.all, ci.bpfree,  ci.ecd, ci.sub01, 
          ct.aki.all, ct.bpfree, ct.ecd, ct.sub01, 
          nrow=4, ncol=4)

ggarrange(ci.all, ci.dcd, ci.bp, ci.sub, 
          ct.all, ct.dcd, ct.bp, ct.sub,
          ci.aki.all, ci.aki.bp, ci.bpfree, ci.sub01, 
          ct.aki.all, ct.aki.bp, ct.bpfree, ct.sub01, 
          nrow=4, ncol=4)

```


```{r}
####################
# PROTOCOL KINETICS - AKI groups (GRAPHS only)
###################

r.set<-dgf_filter[which(dgf_filter$bpar_any==1),]

# plot the i scores across protocol biopsies to 12 months, split by DGF group
i.mean.0<-tapply(r.set$i_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.0<-tapply(r.set$i_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_0m)-sum(is.na(r.set$i_0m)))
i.mean.1<-tapply(r.set$i_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.1<-tapply(r.set$i_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_1m)-sum(is.na(r.set$i_1m)))
i.mean.3<-tapply(r.set$i_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.3<-tapply(r.set$i_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_3m)-sum(is.na(r.set$i_3m)))
i.mean.12<-tapply(r.set$i_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
i.sem.12<-tapply(r.set$i_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$i_12m)-sum(is.na(r.set$i_12m)))

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1,2,3)

protocol.i.mean<- rbind(i.mean.0, i.mean.1, i.mean.3, i.mean.12)
protocol.i.mean<-melt(protocol.i.mean, id.var=dgf)
protocol.i.mean<-cbind(p.time, protocol.i.mean)
colnames(protocol.i.mean)<-c("time", "stat", "strata", "mean")

protocol.i.sem <- rbind(i.sem.0, i.sem.1, i.sem.3, i.sem.12)
protocol.i.sem<-melt(protocol.i.sem, id.var=dgf)
protocol.i.sem<-cbind(p.time, protocol.i.sem)
colnames(protocol.i.sem)<-c("time", "stat", "strata", "sem")

protocol.i<-protocol.i.mean
protocol.i<-as.data.frame(protocol.i)
protocol.i$sem<-protocol.i.sem$sem

# t-score
t.mean.0<-tapply(r.set$t_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.0<-tapply(r.set$t_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_0m)-sum(is.na(r.set$t_0m)))
t.mean.1<-tapply(r.set$t_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.1<-tapply(r.set$t_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_1m)-sum(is.na(r.set$t_1m)))
t.mean.3<-tapply(r.set$t_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.3<-tapply(r.set$t_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_3m)-sum(is.na(r.set$t_3m)))
t.mean.12<-tapply(r.set$t_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
t.sem.12<-tapply(r.set$t_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$t_12m)-sum(is.na(r.set$t_12m)))

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1,2,3)

protocol.t.mean<- rbind(t.mean.0, t.mean.1, t.mean.3, t.mean.12)
protocol.t.mean<-melt(protocol.t.mean,id.var= dgf)
protocol.t.mean<-cbind(p.time, protocol.t.mean)
colnames(protocol.t.mean)<-c("time", "stat", "strata", "mean")

protocol.t.sem <- rbind(t.sem.0, t.sem.1, t.sem.3, t.sem.12)
protocol.t.sem<-melt(protocol.t.sem, id.var=dgf)
protocol.t.sem<-cbind(p.time, protocol.t.sem)
colnames(protocol.t.sem)<-c("time", "stat", "strata", "sem")

protocol.t<-protocol.t.mean
protocol.t<-as.data.frame(protocol.t)
protocol.t$sem<-protocol.t.sem$sem

# ci score
ci.mean.0<-tapply(r.set$ci_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.0<-tapply(r.set$ci_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_0m)-sum(is.na(r.set$ci_0m)))
ci.mean.1<-tapply(r.set$ci_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.1<-tapply(r.set$ci_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_1m)-sum(is.na(r.set$ci_1m)))
ci.mean.3<-tapply(r.set$ci_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.3<-tapply(r.set$ci_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_3m)-sum(is.na(r.set$ci_3m)))
ci.mean.12<-tapply(r.set$ci_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ci.sem.12<-tapply(r.set$ci_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ci_12m)-sum(is.na(r.set$ci_12m)))

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1,2,3)

protocol.ci.mean<- rbind(ci.mean.0, ci.mean.1, ci.mean.3, ci.mean.12)
protocol.ci.mean<-melt(protocol.ci.mean, id.var=dgf)
protocol.ci.mean<-cbind(p.time, protocol.ci.mean)
colnames(protocol.ci.mean)<-c("time", "stat", "strata", "mean")

protocol.ci.sem <- rbind(ci.sem.0, ci.sem.1, ci.sem.3, ci.sem.12)
protocol.ci.sem<-melt(protocol.ci.sem, id.var=dgf)
protocol.ci.sem<-cbind(p.time, protocol.ci.sem)
colnames(protocol.ci.sem)<-c("time", "stat", "strata", "sem")

protocol.ci<-protocol.ci.mean
protocol.ci<-as.data.frame(protocol.ci)
protocol.ci$sem<-protocol.ci.sem$sem

# ct score
ct.mean.0<-tapply(r.set$ct_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.0<-tapply(r.set$ct_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_0m)-sum(is.na(r.set$ct_0m)))
ct.mean.1<-tapply(r.set$ct_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.1<-tapply(r.set$ct_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_1m)-sum(is.na(r.set$ct_1m)))
ct.mean.3<-tapply(r.set$ct_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.3<-tapply(r.set$ct_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_3m)-sum(is.na(r.set$ct_3m)))
ct.mean.12<-tapply(r.set$ct_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
ct.sem.12<-tapply(r.set$ct_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ct_12m)-sum(is.na(r.set$ct_12m)))

p.time<- c(0,1,3,12, 0,1,3,12)
dgf<-c(0,1,2,3)

protocol.ct.mean<- rbind(ct.mean.0, ct.mean.1, ct.mean.3, ct.mean.12)
protocol.ct.mean<-melt(protocol.ct.mean, id.var=dgf)
protocol.ct.mean<-cbind(p.time, protocol.ct.mean)
colnames(protocol.ct.mean)<-c("time", "stat", "strata", "mean")

protocol.ct.sem <- rbind(ct.sem.0, ct.sem.1, ct.sem.3, ct.sem.12)
protocol.ct.sem<-melt(protocol.ct.sem, id.var=dgf)
protocol.ct.sem<-cbind(p.time, protocol.ct.sem)
colnames(protocol.ct.sem)<-c("time", "stat", "strata", "sem")

protocol.ct<-protocol.ct.mean
protocol.ct<-as.data.frame(protocol.ct)
protocol.ct$sem<-protocol.ct.sem$sem

# plots
i.score<-
  ggplot(protocol.i, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
    ylim(0,1.8)+
  ggtitle("Banff-i")+
  xlab("Time (months)")+
  ylab("Banff-i score")

ci.score<-
  ggplot(protocol.ci, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
  ylim(0,1.8)+
  ggtitle("Banff-ci (AKI-BPAR)")+
  xlab("Time (months)")+
  ylab("Banff-ci score")

t.score<-
  ggplot(protocol.t, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
    ylim(0,1.8)+
  ggtitle("Banff-t")+
  xlab("Time (months)")+
  ylab("Banff-t score")

ct.score<-
  ggplot(protocol.ct, aes(x=time, y=mean, group = strata, colour=strata)) + 
  geom_point() +
  geom_path() + 
  geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
  theme_classic() +
  ylim(0,1.8)+
  ggtitle("Banff-ct (AKI-BPAR)")+
  xlab("Time (months)")+
  ylab("Banff-ct score")

```


```{r}
############ OLD ############
# # g-score
# g.mean.0<-tapply(r.set$g_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# g.sem.0<-tapply(r.set$g_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$g_0m)-sum(is.na(r.set$g_0m)))
# g.mean.1<-tapply(r.set$g_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# g.sem.1<-tapply(r.set$g_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$g_1m)-sum(is.na(r.set$g_1m)))
# g.mean.3<-tapply(r.set$g_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# g.sem.3<-tapply(r.set$g_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$g_3m)-sum(is.na(r.set$g_3m)))
# g.mean.12<-tapply(r.set$g_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# g.sem.12<-tapply(r.set$g_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$g_12m)-sum(is.na(r.set$g_12m)))
# 
# gtest1<- t.test(r.set$g_1m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# gtest3<- t.test(r.set$g_3m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# gtest12<- t.test(r.set$g_12m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# gtest0<- t.test(r.set$g_0m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# g.ttest<-c(gtest0$p.value, gtest1$p.value, gtest3$p.value, gtest12$p.value)
# 
# p.time<- c(0,1,3,12, 0,1,3,12)
# dgf<-c(0,1)

# protocol.g.mean<- rbind(g.mean.0, g.mean.1, g.mean.3, g.mean.12)
# protocol.g.mean<-melt(protocol.g.mean, dgf)
# protocol.g.mean<-cbind(p.time, protocol.g.mean)
# colnames(protocol.g.mean)<-c("time", "stat", "strata", "mean")
# 
# protocol.g.sem <- rbind(g.sem.0, g.sem.1, g.sem.3, g.sem.12)
# protocol.g.sem<-melt(protocol.g.sem, dgf)
# protocol.g.sem<-cbind(p.time, protocol.g.sem)
# colnames(protocol.g.sem)<-c("time", "stat", "strata", "sem")
# 
# protocol.g<-protocol.g.mean
# protocol.g<-as.data.frame(protocol.g)
# protocol.g$sem<-protocol.g.sem$sem
# 
# # v-score
# v.mean.0<-tapply(r.set$v_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# v.sem.0<-tapply(r.set$v_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$v_0m)-sum(is.na(r.set$v_0m)))
# v.mean.1<-tapply(r.set$v_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# v.sem.1<-tapply(r.set$v_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$v_1m)-sum(is.na(r.set$v_1m)))
# v.mean.3<-tapply(r.set$v_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# v.sem.3<-tapply(r.set$v_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$v_3m)-sum(is.na(r.set$v_3m)))
# v.mean.12<-tapply(r.set$v_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# v.sem.12<-tapply(r.set$v_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$v_12m)-sum(is.na(r.set$v_12m)))
# 
# vtest1<- t.test(r.set$v_1m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# vtest3<- t.test(r.set$v_3m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# vtest12<- t.test(r.set$v_12m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# vtest0<- t.test(r.set$v_0m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# v.ttest<-c(vtest0$p.value, vtest1$p.value, vtest3$p.value, vtest12$p.value)
# 
# p.time<- c(0,1,3,12, 0,1,3,12)
# dgf<-c(0,1)

# protocol.v.mean<- rbind(v.mean.0, v.mean.1, v.mean.3, v.mean.12)
# protocol.v.mean<-melt(protocol.v.mean, dgf)
# protocol.v.mean<-cbind(p.time, protocol.v.mean)
# colnames(protocol.v.mean)<-c("time", "stat", "strata", "mean")
# 
# protocol.v.sem <- rbind(v.sem.0, v.sem.1, v.sem.3, v.sem.12)
# protocol.v.sem<-melt(protocol.v.sem, dgf)
# protocol.v.sem<-cbind(p.time, protocol.v.sem)
# colnames(protocol.v.sem)<-c("time", "stat", "strata", "sem")
# 
# protocol.v<-protocol.v.mean
# protocol.v<-as.data.frame(protocol.v)
# protocol.v$sem<-protocol.v.sem$sem
# # cg-score
# cg.mean.0<-tapply(r.set$cg_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cg.sem.0<-tapply(r.set$cg_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cg_0m)-sum(is.na(r.set$cg_0m)))
# cg.mean.1<-tapply(r.set$cg_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cg.sem.1<-tapply(r.set$cg_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cg_1m)-sum(is.na(r.set$cg_1m)))
# cg.mean.3<-tapply(r.set$cg_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cg.sem.3<-tapply(r.set$cg_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cg_3m)-sum(is.na(r.set$cg_3m)))
# cg.mean.12<-tapply(r.set$cg_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cg.sem.12<-tapply(r.set$cg_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cg_12m)-sum(is.na(r.set$cg_12m)))
# 
# cgtest1<- t.test(r.set$cg_1m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cgtest3<- t.test(r.set$cg_3m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cgtest12<- t.test(r.set$cg_12m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cgtest0<- t.test(r.set$cg_0m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cg.ttest<-c(cgtest0$p.value, cgtest1$p.value, cgtest3$p.value, cgtest12$p.value)
# 
# p.time<- c(0,1,3,12, 0,1,3,12)
# dgf<-c(0,1)
# 
# protocol.cg.mean<- rbind(cg.mean.0, cg.mean.1, cg.mean.3, cg.mean.12)
# protocol.cg.mean<-melt(protocol.cg.mean, dgf)
# protocol.cg.mean<-cbind(p.time, protocol.cg.mean)
# colnames(protocol.cg.mean)<-c("time", "stat", "strata", "mean")
# 
# protocol.cg.sem <- rbind(cg.sem.0, cg.sem.1, cg.sem.3, cg.sem.12)
# protocol.cg.sem<-melt(protocol.cg.sem, dgf)
# protocol.cg.sem<-cbind(p.time, protocol.cg.sem)
# colnames(protocol.cg.sem)<-c("time", "stat", "strata", "sem")
# 
# protocol.cg<-protocol.cg.mean
# protocol.cg<-as.data.frame(protocol.cg)
# protocol.cg$sem<-protocol.cg.sem$sem
# 
# # cv-score
# cv.mean.0<-tapply(r.set$cv_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cv.sem.0<-tapply(r.set$cv_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cv_0m)-sum(is.na(r.set$cv_0m)))
# cv.mean.1<-tapply(r.set$cv_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cv.sem.1<-tapply(r.set$cv_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cv_1m)-sum(is.na(r.set$cv_1m)))
# cv.mean.3<-tapply(r.set$cv_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cv.sem.3<-tapply(r.set$cv_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cv_3m)-sum(is.na(r.set$cv_3m)))
# cv.mean.12<-tapply(r.set$cv_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# cv.sem.12<-tapply(r.set$cv_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$cv_12m)-sum(is.na(r.set$cv_12m)))
# 
# cvtest1<- t.test(r.set$cv_1m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cvtest3<- t.test(r.set$cv_3m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cvtest12<- t.test(r.set$cv_12m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cvtest0<- t.test(r.set$cv_0m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# cv.ttest<-c(cvtest0$p.value, cvtest1$p.value, cvtest3$p.value, cvtest12$p.value)
# 
# p.time<- c(0,1,3,12, 0,1,3,12)
# dgf<-c(0,1)
# 
# protocol.cv.mean<- rbind(cv.mean.0, cv.mean.1, cv.mean.3, cv.mean.12)
# protocol.cv.mean<-melt(protocol.cv.mean, dgf)
# protocol.cv.mean<-cbind(p.time, protocol.cv.mean)
# colnames(protocol.cv.mean)<-c("time", "stat", "strata", "mean")
# 
# protocol.cv.sem <- rbind(cv.sem.0, cv.sem.1, cv.sem.3, cv.sem.12)
# protocol.cv.sem<-melt(protocol.cv.sem, dgf)
# protocol.cv.sem<-cbind(p.time, protocol.cv.sem)
# colnames(protocol.cv.sem)<-c("time", "stat", "strata", "sem")
# 
# protocol.cv<-protocol.cv.mean
# protocol.cv<-as.data.frame(protocol.cv)
# protocol.cv$sem<-protocol.cv.sem$sem
# 
# # ah-score
# ah.mean.0<-tapply(r.set$ah_0m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# ah.sem.0<-tapply(r.set$ah_0m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ah_0m)-sum(is.na(r.set$ah_0m)))
# ah.mean.1<-tapply(r.set$ah_1m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# ah.sem.1<-tapply(r.set$ah_1m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ah_1m)-sum(is.na(r.set$ah_1m)))
# ah.mean.3<-tapply(r.set$ah_3m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# ah.sem.3<-tapply(r.set$ah_3m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ah_3m)-sum(is.na(r.set$ah_3m)))
# ah.mean.12<-tapply(r.set$ah_12m, r.set$dgf_criteria1, mean, na.rm=TRUE)
# ah.sem.12<-tapply(r.set$ah_12m, r.set$dgf_criteria1, sd, na.rm=TRUE)/sqrt(length(r.set$ah_12m)-sum(is.na(r.set$ah_12m)))
# 
# ahtest1<- t.test(r.set$ah_1m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# ahtest3<- t.test(r.set$ah_3m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# ahtest12<- t.test(r.set$ah_12m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# ahtest0<- t.test(r.set$ah_0m, r.set$dgf_criteria, alternative = "two.sided", var.equal = FALSE)
# ah.ttest<-c(ahtest0$p.value, ahtest1$p.value, ahtest3$p.value, ahtest12$p.value)
# 
# p.time<- c(0,1,3,12, 0,1,3,12)
# dgf<-c(0,1)
# 
# protocol.ah.mean<- rbind(ah.mean.0, ah.mean.1, ah.mean.3, ah.mean.12)
# protocol.ah.mean<-melt(protocol.ah.mean, dgf)
# protocol.ah.mean<-cbind(p.time, protocol.ah.mean)
# colnames(protocol.ah.mean)<-c("time", "stat", "strata", "mean")
# 
# protocol.ah.sem <- rbind(ah.sem.0, ah.sem.1, ah.sem.3, ah.sem.12)
# protocol.ah.sem<-melt(protocol.ah.sem, dgf)
# protocol.ah.sem<-cbind(p.time, protocol.ah.sem)
# colnames(protocol.ah.sem)<-c("time", "stat", "strata", "sem")
# 
# protocol.ah<-protocol.ah.mean
# protocol.ah<-as.data.frame(protocol.ah)
# protocol.ah$sem<-protocol.ah.sem$sem

# pvalues<-cbind(i.ttest, t.ttest, ci.ttest, ct.ttest)
# # g.ttest, v.ttest, ah.ttest, , cg.ttest, cv.ttest
# rownames(pvalues)<-c("0m", "1m", "3m", "12m")
# pvalues<-round(pvalues, digits = 3)
# View(pvalues)

# g.score<-
#   ggplot(protocol.g, aes(x=time, y=mean, group = strata, colour=strata)) + 
#   geom_point() +
#   geom_path() + 
#   geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
#   theme_classic() +
#   ggtitle("Banff-g")+
#   xlab("Time (months)")+
#   ylab("Banff-g score")
# 
# cg.score<-
#   ggplot(protocol.cg, aes(x=time, y=mean, group = strata, colour=strata)) + 
#   geom_point() +
#   geom_path() + 
#   geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
#   theme_classic() +
#   ggtitle("Banff-cg")+
#   xlab("Time (months)")+
#   ylab("Banff-cg score")
# 
# v.score<-
#   ggplot(protocol.v, aes(x=time, y=mean, group = strata, colour=strata)) + 
#   geom_point() +
#   geom_path() + 
#   geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
#   theme_classic() +
#   ggtitle("Banff-v")+
#   xlab("Time (months)")+
#   ylab("Banff-v score")
# 
# cv.score<-
#   ggplot(protocol.cv, aes(x=time, y=mean, group = strata, colour=strata)) + 
#   geom_point() +
#   geom_path() + 
#   geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
#   theme_classic() +
#   ggtitle("Banff-cv")+
#   xlab("Time (months)")+
#   ylab("Banff-cv score")
# 
# ah.score<-
#   ggplot(protocol.ah, aes(x=time, y=mean, group = strata, colour=strata)) + 
#   geom_point() +
#   geom_path() + 
#   geom_errorbar(aes(ymin=(mean-sem), ymax=(mean+sem), group=strata),  width=.2) +
#   theme_classic() +
#   ggtitle("Banff-ah")+
#   xlab("Time (months)")+
#   ylab("Banff-ah score")
# ggarrange(i.score, t.score, g.score, v.score, ci.score, ct.score, cg.score, cv.score, ah.score)

```


```{r}
# # Make a list to define the biopsies in sequential order
# # sort by dates, check excel to R import hasn't changed to numeric, in our set - only 3m and 3y needs correction
#dgf_filter$protocol_3m_date<-as.numeric(dgf_filter$protocol_3m_date)
#dgf_filter$protocol_3m_date<-as.Date(dgf_filter$protocol_3m_date, origin = "1899-12-30")
# 
# dgf_filter$protocol_3y_date<-as.numeric(dgf_filter$protocol_3y_date)
# dgf_filter$protocol_3y_date<-as.Date(dgf_filter$protocol_3y_date, origin = "1899-12-30")
# 
# # make new variable for protocols using ibx so it's compatible
# dgf_filter$date_ibx0mp <- dgf_filter$protocol_0m_date
# dgf_filter$date_ibx1mp <- dgf_filter$protocol_1m_date
# dgf_filter$date_ibx3mp <- dgf_filter$protocol_3m_date
# dgf_filter$date_ibx12mp <- dgf_filter$protocol_12m_date
# dgf_filter$date_ibx3yp <- dgf_filter$protocol_3y_date
# dgf_filter$date_ibx5yp <- dgf_filter$protocol_5y_date
# 
# bxdates<-dgf_filter[, c("study_ID", "date_ibx0mp", "date_ibx1mp", "date_ibx3mp", "date_ibx12mp", "date_ibx3yp", "date_ibx5yp", "date_ibx1", "date_ibx2", "date_ibx3", "date_ibx4", "date_ibx5", "date_ibx6", "date_ibx7")]
# 
# # for strategy, just apply to row 1 then make into for loop using min(study_ID): max(study_ID)
# 
# # select row and then sort
# bxdates_r1 <- bxdates[5,]
# 
# bxdates_r2 <-t(bxdates_r1)
# bxdates_r2 <- bxdates_r2[!is.na(bxdates_r2),]
# sorted_r2<-sort(as.Date(bxdates_r2[c(2:length(bxdates_r2))]))
# sorted <-rbind(bxdates_r2[1,1], sorted_r2)
# 
# bxdates_output <- t(bxdates_r1)
# bxdates_output <- ifelse(is.na(bxdates_output)==FALSE, NA, NA)
# bxdates_output <- merge(bxdates_output, sorted_r2)
# 
# View(bxdates_r2)

# sorted_r2 <- order(as.Date(bxdates_r2[c(2:length(bxdates_r2))]))
# sorted_r2 <- c(NA, sorted_r2)
# bxdates_r2$order <-sorted_r2
# label(bxdates_r2)<-c("dates", "order")
# bxdates_r2<- arrange(bxdates_r2, group_by(bxdates_r2$order))

# find out how many biopsies (with scoring) done for each patient, label yes = 1, no = 0
# dgf_filter$bx0m <- ifelse (is.na(dgf_filter$protocol_0m_date)==FALSE & is.na(dgf_filter$i_0m)==FALSE,1,0)
# dgf_filter$bx1m <- ifelse (is.na(dgf_filter$protocol_1m_date)==FALSE & is.na(dgf_filter$i_1m)==FALSE,1,0)
# dgf_filter$bx3m <- ifelse (is.na(dgf_filter$protocol_3m_date)==FALSE & is.na(dgf_filter$i_3m)==FALSE,1,0)
# dgf_filter$bx12m <- ifelse (is.na(dgf_filter$protocol_12m_date)==FALSE & is.na(dgf_filter$i_12m)==FALSE,1,0)
# dgf_filter$bx3y <- ifelse (is.na(dgf_filter$protocol_3y_date)==FALSE & is.na(dgf_filter$i_3y)==FALSE,1,0)
# dgf_filter$bx5y <- ifelse (is.na(dgf_filter$protocol_5y_date)==FALSE & is.na(dgf_filter$i_5y)==FALSE,1,0)
# 
# dgf_filter$bxind1 <- ifelse (is.na(dgf_filter$i_ibx1)==FALSE & is.na(dgf_filter$date_ibx1)==FALSE,1,0)
# dgf_filter$bxind2 <- ifelse (is.na(dgf_filter$i_ibx2)==FALSE & is.na(dgf_filter$date_ibx2)==FALSE,1,0)
# dgf_filter$bxind3 <- ifelse (is.na(dgf_filter$i_ibx3)==FALSE & is.na(dgf_filter$date_ibx3)==FALSE,1,0)
# dgf_filter$bxind4 <- ifelse (is.na(dgf_filter$i_ibx4)==FALSE & is.na(dgf_filter$date_ibx4)==FALSE,1,0)
# dgf_filter$bxind5 <- ifelse (is.na(dgf_filter$i_ibx5)==FALSE & is.na(dgf_filter$date_ibx5)==FALSE,1,0)
# dgf_filter$bxind6 <- ifelse (is.na(dgf_filter$i_ibx6)==FALSE & is.na(dgf_filter$date_ibx6)==FALSE,1,0)
# dgf_filter$bxind7 <- ifelse (is.na(dgf_filter$i_ibx7)==FALSE & is.na(dgf_filter$date_ibx7)==FALSE,1,0)
# 
# dgf_filter$bxtotal <- dgf_filter$bx0m + dgf_filter$bx1m + dgf_filter$bx3m + dgf_filter$bx12m + + dgf_filter$bx3y + dgf_filter$bx5y + dgf_filter$bxind1 +dgf_filter$bxind2 +dgf_filter$bxind3 +dgf_filter$bxind4 +dgf_filter$bxind5 + dgf_filter$bxind6 + dgf_filter$bxind7
# 
# View(dgf_filter$bxtotal)

# # group timepoint/patient
# bx0m <-dgf_filter[,c("study_ID", "protocol_0m_date", "gloms_0m", "sclerosed_0m", "i_0m", "t_0m", "ti_0m", "g_0m", "v_0m", "ptc_0m", "ah_0m", "mm_0m", "ci_0m", "ct_0m", "cg_0m", "cv_0m", "c4d_0m", "sv40_0m", "comment_0m")] # add IFTA for later months
# bx0m$order<-NA
# 
# bx1m <-dgf_filter[,c("study_ID", "protocol_1m_date", "gloms_1m", "sclerosed_1m", "i_1m", "t_1m", "ti_1m", "g_1m", "v_1m", "ptc_1m", "ah_1m", "mm_1m", "ci_0m", "ct_0m", "cg_0m", "cv_0m", "c4d_0m", "sv40_0m", "iIFTA_1m", "tIFTA_1m", "comment_0m")] 
# bx1m$order<-NA
# 
# bx3m <-dgf_filter[,c("study_ID", "protocol_3m_date", "gloms_3m", "sclerosed_3m", "i_3m", "t_3m", "ti_3m", "g_3m", "v_3m", "ptc_3m", "ah_3m", "mm_3m", "ci_3m", "ct_3m", "cg_3m", "cv_3m", "c4d_3m", "sv40_3m","iIFTA_3m", "tIFTA_3m", "comment_3m")] 
# bx3m$order<-NA
# 
# bx12m <-dgf_filter[,c("study_ID", "protocol_12m_date", "gloms_12m", "sclerosed_12m", "i_12m", "t_12m", "ti_12m", "g_12m", "v_12m", "ptc_12m", "ah_12m", "mm_12m", "ci_12m", "ct_12m", "cg_12m", "cv_12m", "c4d_12m", "sv40_12m","iIFTA_12m", "tIFTA_12m", "comment_12m")] 
# bx12m$order<-NA
# 
# bx3y <-dgf_filter[,c("study_ID", "protocol_3y_date", "gloms_3y", "sclerosed_3y", "i_3y", "t_3y", "ti_3y", "g_3y", "v_3y", "ptc_3y", "ah_3y", "mm_3y", "ci_3y", "ct_3y", "cg_3y", "cv_3y", "c4d_3y", "sv40_3y","iIFTA_3y", "tIFTA_3y", "comment_3y")] 
# bx3y$order<-NA
# 
# bx5y <-dgf_filter[,c("study_ID", "protocol_5y_date", "gloms_5y", "sclerosed_5y", "i_5y", "t_5y", "ti_5y", "g_5y", "v_5y", "ptc_5y", "ah_5y", "mm_5y", "ci_5y", "ct_5y", "cg_5y", "cv_5y", "c4d_5y", "sv40_5y","iIFTA_5y", "tIFTA_5y", "comment_5y")] 
# bx5y$order<-NA
# 
# bxibx1 <-dgf_filter[,c("study_ID", "date_ibx1",  "gloms_ibx1", "sclerosed_ibx1", "i_ibx1", "t_ibx1", "ti_ibx1", "g_ibx1", "v_ibx1", "ptc_ibx1", "ah_ibx1", "mm_ibx1", "ci_ibx1", "ct_ibx1", "cg_ibx1", "cv_ibx1", "c4d_ibx1", "sv40_ibx1","iIFTA_ibx1", "tIFTA_ibx1", "comment_ibx1")] 
# bxibx1$order<-NA
# 
# bxibx2 <-dgf_filter[,c("study_ID", "date_ibx2",  "gloms_ibx2", "sclerosed_ibx2", "i_ibx2", "t_ibx2", "ti_ibx2", "g_ibx2", "v_ibx2", "ptc_ibx2", "ah_ibx2", "mm_ibx2", "ci_ibx2", "ct_ibx2", "cg_ibx2", "cv_ibx2", "c4d_ibx2", "sv40_ibx2","iIFTA_ibx2", "tIFTA_ibx2", "comment_ibx2")] 
# bxibx2$order<-NA
# 
# bxibx3 <-dgf_filter[,c("study_ID","date_ibx3",  "gloms_ibx3", "sclerosed_ibx3", "i_ibx3", "t_ibx3", "ti_ibx3", "g_ibx3", "v_ibx3", "ptc_ibx3", "ah_ibx3", "mm_ibx3", "ci_ibx3", "ct_ibx3", "cg_ibx3", "cv_ibx3", "c4d_ibx3", "sv40_ibx3","iIFTA_ibx3", "tIFTA_ibx3", "comment_ibx3")] 
# bxibx3$order<-NA
# 
# bxibx4 <-dgf_filter[,c("study_ID", "date_ibx4",  "gloms_ibx4", "sclerosed_ibx4", "i_ibx4", "t_ibx4", "ti_ibx4", "g_ibx4", "v_ibx4", "ptc_ibx4", "ah_ibx4", "mm_ibx4", "ci_ibx4", "ct_ibx4", "cg_ibx4", "cv_ibx4", "c4d_ibx4", "sv40_ibx4","iIFTA_ibx4", "tIFTA_ibx4", "comment_ibx4")] 
# bxibx4$order<-NA
# 
# bxibx5 <-dgf_filter[,c("study_ID", "date_ibx5",  "gloms_ibx5", "sclerosed_ibx5", "i_ibx5", "t_ibx5", "ti_ibx5", "g_ibx5", "v_ibx5", "ptc_ibx5", "ah_ibx5", "mm_ibx5", "ci_ibx5", "ct_ibx5", "cg_ibx5", "cv_ibx5", "c4d_ibx5", "sv40_ibx5","iIFTA_ibx5", "tIFTA_ibx5", "comment_ibx5")] 
# bxibx5$order<-NA
# 
# bxibx6 <-dgf_filter[,c("study_ID", "date_ibx6",  "gloms_ibx6", "sclerosed_ibx6", "i_ibx6", "t_ibx6", "ti_ibx6", "g_ibx6", "v_ibx6", "ptc_ibx6", "ah_ibx6", "mm_ibx6", "ci_ibx6", "ct_ibx6", "cg_ibx6", "cv_ibx6", "c4d_ibx6", "sv40_ibx6","iIFTA_ibx6", "tIFTA_ibx6", "comment_ibx1")] 
# bxibx6$order<-NA
# 
# bxibx7 <-dgf_filter[,c("study_ID", "date_ibx7",  "gloms_ibx7", "sclerosed_ibx7", "i_ibx7", "t_ibx7", "ti_ibx7", "g_ibx7", "v_ibx7", "ptc_ibx7", "ah_ibx7", "mm_ibx7", "ci_ibx7", "ct_ibx7", "cg_ibx7", "cv_ibx7", "c4d_ibx7", "sv40_ibx7","iIFTA_ibx7", "tIFTA_ibx7", "comment_ibx7")] 
# bxibx7$order<-NA

# sprt, pair, then group
```
















